From 07d8fcf3f6182a62782b7d4dec2a9fe4ab803747 Mon Sep 17 00:00:00 2001
From: Louis Lenders <xerox.xerox2000x@gmail.com>
Date: Tue, 18 Mar 2025 18:54:18 +0100
Subject: [PATCH 249/327] win32u: Add stub for
 NtUserSetAdditionalForegroundBoostProcesses.

Microsoft Edge worked if windows version is set to win8.1 (or win7) but
now crashes in current wine due to this unimplemented function.
---
 dlls/msvcp_win/msvcp_win.spec |   6 +-
 dlls/ntdll/ntsyscalls.h       | 503 +++++++++++++++++++
 dlls/user32/user32.spec       |  26 +-
 dlls/win32u/main.c            |   5 +
 dlls/win32u/sysparams.c       |  10 +
 dlls/win32u/win32syscalls.h   | 891 ++++++++++++++++++++++++++++++++++
 dlls/win32u/win32u.spec       |   2 +-
 dlls/wow64win/user.c          |  19 +
 include/ntuser.h              |   1 +
 9 files changed, 1446 insertions(+), 17 deletions(-)
 create mode 100644 dlls/ntdll/ntsyscalls.h
 create mode 100644 dlls/win32u/win32syscalls.h

diff --git a/dlls/msvcp_win/msvcp_win.spec b/dlls/msvcp_win/msvcp_win.spec
index f89a280f4ca..3b53cb8b564 100644
--- a/dlls/msvcp_win/msvcp_win.spec
+++ b/dlls/msvcp_win/msvcp_win.spec
@@ -175,9 +175,9 @@
 @ cdecl -arch=arm ??0?$codecvt@_SDU_Mbstatet@@@std@@QAA@ABV_Locinfo@1@I@Z(ptr ptr long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QAA@ABV_Locinfo@1@I@Z
 @ thiscall -arch=i386 ??0?$codecvt@_SDU_Mbstatet@@@std@@QAE@ABV_Locinfo@1@I@Z(ptr ptr long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QAE@ABV_Locinfo@1@I@Z
 @ cdecl -arch=win64 ??0?$codecvt@_SDU_Mbstatet@@@std@@QEAA@AEBV_Locinfo@1@_K@Z(ptr ptr long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QEAA@AEBV_Locinfo@1@_K@Z
-@ cdecl -arch=arm ??0?$codecvt@_SDU_Mbstatet@@@std@@QAA@ABV_Locinfo@1@KW4_Codecvt_mode@1@I@Z(ptr long long long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QAA@ABV_Locinfo@1@KW4_Codecvt_mode@1@I@Z
-@ thiscall -arch=i386 ??0?$codecvt@_SDU_Mbstatet@@@std@@QAE@ABV_Locinfo@1@KW4_Codecvt_mode@1@I@Z(ptr long long long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QAE@ABV_Locinfo@1@KW4_Codecvt_mode@1@I@Z
-@ cdecl -arch=win64 ??0?$codecvt@_SDU_Mbstatet@@@std@@QEAA@AEBV_Locinfo@1@KW4_Codecvt_mode@1@_K@Z(ptr long long long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QEAA@AEBV_Locinfo@1@KW4_Codecvt_mode@1@_K@Z
+@ cdecl -arch=arm ??0?$codecvt@_SDU_Mbstatet@@@std@@QAA@ABV_Locinfo@1@KW4_Codecvt_mode@1@I@Z(ptr ptr long long long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QAA@ABV_Locinfo@1@KW4_Codecvt_mode@1@I@Z
+@ thiscall -arch=i386 ??0?$codecvt@_SDU_Mbstatet@@@std@@QAE@ABV_Locinfo@1@KW4_Codecvt_mode@1@I@Z(ptr ptr long long long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QAE@ABV_Locinfo@1@KW4_Codecvt_mode@1@I@Z
+@ cdecl -arch=win64 ??0?$codecvt@_SDU_Mbstatet@@@std@@QEAA@AEBV_Locinfo@1@KW4_Codecvt_mode@1@_K@Z(ptr ptr long long long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QEAA@AEBV_Locinfo@1@KW4_Codecvt_mode@1@_K@Z
 @ cdecl -arch=arm ??0?$codecvt@_SDU_Mbstatet@@@std@@QAA@I@Z(ptr long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QAA@I@Z
 @ thiscall -arch=i386 ??0?$codecvt@_SDU_Mbstatet@@@std@@QAE@I@Z(ptr long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QAE@I@Z
 @ cdecl -arch=win64 ??0?$codecvt@_SDU_Mbstatet@@@std@@QEAA@_K@Z(ptr long) msvcp140.??0?$codecvt@_SDU_Mbstatet@@@std@@QEAA@_K@Z
diff --git a/dlls/ntdll/ntsyscalls.h b/dlls/ntdll/ntsyscalls.h
new file mode 100644
index 00000000000..abfb5a73805
--- /dev/null
+++ b/dlls/ntdll/ntsyscalls.h
@@ -0,0 +1,503 @@
+/* Automatically generated by tools/make_specfiles */
+
+#define ALL_SYSCALLS32 \
+    SYSCALL_ENTRY( 0x0000, NtAcceptConnectPort, 24 ) \
+    SYSCALL_ENTRY( 0x0001, NtAccessCheck, 32 ) \
+    SYSCALL_ENTRY( 0x0002, NtAccessCheckAndAuditAlarm, 44 ) \
+    SYSCALL_ENTRY( 0x0003, NtAddAtom, 12 ) \
+    SYSCALL_ENTRY( 0x0004, NtAdjustGroupsToken, 24 ) \
+    SYSCALL_ENTRY( 0x0005, NtAdjustPrivilegesToken, 24 ) \
+    SYSCALL_ENTRY( 0x0006, NtAlertResumeThread, 8 ) \
+    SYSCALL_ENTRY( 0x0007, NtAlertThread, 4 ) \
+    SYSCALL_ENTRY( 0x0008, NtAlertThreadByThreadId, 4 ) \
+    SYSCALL_ENTRY( 0x0009, NtAllocateLocallyUniqueId, 4 ) \
+    SYSCALL_ENTRY( 0x000a, NtAllocateReserveObject, 12 ) \
+    SYSCALL_ENTRY( 0x000b, NtAllocateUuids, 16 ) \
+    SYSCALL_ENTRY( 0x000c, NtAllocateVirtualMemory, 24 ) \
+    SYSCALL_ENTRY( 0x000d, NtAllocateVirtualMemoryEx, 28 ) \
+    SYSCALL_ENTRY( 0x000e, NtAreMappedFilesTheSame, 8 ) \
+    SYSCALL_ENTRY( 0x000f, NtAssignProcessToJobObject, 8 ) \
+    SYSCALL_ENTRY( 0x0010, NtCallbackReturn, 12 ) \
+    SYSCALL_ENTRY( 0x0011, NtCancelIoFile, 8 ) \
+    SYSCALL_ENTRY( 0x0012, NtCancelIoFileEx, 12 ) \
+    SYSCALL_ENTRY( 0x0013, NtCancelSynchronousIoFile, 12 ) \
+    SYSCALL_ENTRY( 0x0014, NtCancelTimer, 8 ) \
+    SYSCALL_ENTRY( 0x0015, NtClearEvent, 4 ) \
+    SYSCALL_ENTRY( 0x0016, NtClose, 4 ) \
+    SYSCALL_ENTRY( 0x0017, NtCommitTransaction, 8 ) \
+    SYSCALL_ENTRY( 0x0018, NtCompareObjects, 8 ) \
+    SYSCALL_ENTRY( 0x0019, NtCompareTokens, 12 ) \
+    SYSCALL_ENTRY( 0x001a, NtCompleteConnectPort, 4 ) \
+    SYSCALL_ENTRY( 0x001b, NtConnectPort, 32 ) \
+    SYSCALL_ENTRY( 0x001c, NtContinue, 8 ) \
+    SYSCALL_ENTRY( 0x001d, NtContinueEx, 8 ) \
+    SYSCALL_ENTRY( 0x001e, NtConvertBetweenAuxiliaryCounterAndPerformanceCounter, 16 ) \
+    SYSCALL_ENTRY( 0x001f, NtCreateDebugObject, 16 ) \
+    SYSCALL_ENTRY( 0x0020, NtCreateDirectoryObject, 12 ) \
+    SYSCALL_ENTRY( 0x0021, NtCreateEvent, 20 ) \
+    SYSCALL_ENTRY( 0x0022, NtCreateFile, 44 ) \
+    SYSCALL_ENTRY( 0x0023, NtCreateIoCompletion, 16 ) \
+    SYSCALL_ENTRY( 0x0024, NtCreateJobObject, 12 ) \
+    SYSCALL_ENTRY( 0x0025, NtCreateKey, 28 ) \
+    SYSCALL_ENTRY( 0x0026, NtCreateKeyTransacted, 32 ) \
+    SYSCALL_ENTRY( 0x0027, NtCreateKeyedEvent, 16 ) \
+    SYSCALL_ENTRY( 0x0028, NtCreateLowBoxToken, 36 ) \
+    SYSCALL_ENTRY( 0x0029, NtCreateMailslotFile, 32 ) \
+    SYSCALL_ENTRY( 0x002a, NtCreateMutant, 16 ) \
+    SYSCALL_ENTRY( 0x002b, NtCreateNamedPipeFile, 56 ) \
+    SYSCALL_ENTRY( 0x002c, NtCreatePagingFile, 16 ) \
+    SYSCALL_ENTRY( 0x002d, NtCreatePort, 20 ) \
+    SYSCALL_ENTRY( 0x002e, NtCreateSection, 28 ) \
+    SYSCALL_ENTRY( 0x002f, NtCreateSectionEx, 36 ) \
+    SYSCALL_ENTRY( 0x0030, NtCreateSemaphore, 20 ) \
+    SYSCALL_ENTRY( 0x0031, NtCreateSymbolicLinkObject, 16 ) \
+    SYSCALL_ENTRY( 0x0032, NtCreateThread, 32 ) \
+    SYSCALL_ENTRY( 0x0033, NtCreateThreadEx, 44 ) \
+    SYSCALL_ENTRY( 0x0034, NtCreateTimer, 16 ) \
+    SYSCALL_ENTRY( 0x0035, NtCreateToken, 52 ) \
+    SYSCALL_ENTRY( 0x0036, NtCreateTransaction, 40 ) \
+    SYSCALL_ENTRY( 0x0037, NtCreateUserProcess, 44 ) \
+    SYSCALL_ENTRY( 0x0038, NtDebugActiveProcess, 8 ) \
+    SYSCALL_ENTRY( 0x0039, NtDebugContinue, 12 ) \
+    SYSCALL_ENTRY( 0x003a, NtDelayExecution, 8 ) \
+    SYSCALL_ENTRY( 0x003b, NtDeleteAtom, 4 ) \
+    SYSCALL_ENTRY( 0x003c, NtDeleteFile, 4 ) \
+    SYSCALL_ENTRY( 0x003d, NtDeleteKey, 4 ) \
+    SYSCALL_ENTRY( 0x003e, NtDeleteValueKey, 8 ) \
+    SYSCALL_ENTRY( 0x003f, NtDeviceIoControlFile, 40 ) \
+    SYSCALL_ENTRY( 0x0040, NtDisplayString, 4 ) \
+    SYSCALL_ENTRY( 0x0041, NtDuplicateObject, 28 ) \
+    SYSCALL_ENTRY( 0x0042, NtDuplicateToken, 24 ) \
+    SYSCALL_ENTRY( 0x0043, NtEnumerateKey, 24 ) \
+    SYSCALL_ENTRY( 0x0044, NtEnumerateValueKey, 24 ) \
+    SYSCALL_ENTRY( 0x0045, NtFilterToken, 24 ) \
+    SYSCALL_ENTRY( 0x0046, NtFindAtom, 12 ) \
+    SYSCALL_ENTRY( 0x0047, NtFlushBuffersFile, 8 ) \
+    SYSCALL_ENTRY( 0x0048, NtFlushBuffersFileEx, 20 ) \
+    SYSCALL_ENTRY( 0x0049, NtFlushInstructionCache, 12 ) \
+    SYSCALL_ENTRY( 0x004a, NtFlushKey, 4 ) \
+    SYSCALL_ENTRY( 0x004b, NtFlushProcessWriteBuffers, 0 ) \
+    SYSCALL_ENTRY( 0x004c, NtFlushVirtualMemory, 16 ) \
+    SYSCALL_ENTRY( 0x004d, NtFreeVirtualMemory, 16 ) \
+    SYSCALL_ENTRY( 0x004e, NtFsControlFile, 40 ) \
+    SYSCALL_ENTRY( 0x004f, NtGetContextThread, 8 ) \
+    SYSCALL_ENTRY( 0x0050, NtGetCurrentProcessorNumber, 0 ) \
+    SYSCALL_ENTRY( 0x0051, NtGetNextThread, 24 ) \
+    SYSCALL_ENTRY( 0x0052, NtGetNlsSectionPtr, 20 ) \
+    SYSCALL_ENTRY( 0x0053, NtGetWriteWatch, 28 ) \
+    SYSCALL_ENTRY( 0x0054, NtImpersonateAnonymousToken, 4 ) \
+    SYSCALL_ENTRY( 0x0055, NtInitializeNlsFiles, 12 ) \
+    SYSCALL_ENTRY( 0x0056, NtInitiatePowerAction, 16 ) \
+    SYSCALL_ENTRY( 0x0057, NtIsProcessInJob, 8 ) \
+    SYSCALL_ENTRY( 0x0058, NtListenPort, 8 ) \
+    SYSCALL_ENTRY( 0x0059, NtLoadDriver, 4 ) \
+    SYSCALL_ENTRY( 0x005a, NtLoadKey, 8 ) \
+    SYSCALL_ENTRY( 0x005b, NtLoadKey2, 12 ) \
+    SYSCALL_ENTRY( 0x005c, NtLoadKeyEx, 32 ) \
+    SYSCALL_ENTRY( 0x005d, NtLockFile, 40 ) \
+    SYSCALL_ENTRY( 0x005e, NtLockVirtualMemory, 16 ) \
+    SYSCALL_ENTRY( 0x005f, NtMakePermanentObject, 4 ) \
+    SYSCALL_ENTRY( 0x0060, NtMakeTemporaryObject, 4 ) \
+    SYSCALL_ENTRY( 0x0061, NtMapViewOfSection, 40 ) \
+    SYSCALL_ENTRY( 0x0062, NtMapViewOfSectionEx, 36 ) \
+    SYSCALL_ENTRY( 0x0063, NtNotifyChangeDirectoryFile, 36 ) \
+    SYSCALL_ENTRY( 0x0064, NtNotifyChangeKey, 40 ) \
+    SYSCALL_ENTRY( 0x0065, NtNotifyChangeMultipleKeys, 48 ) \
+    SYSCALL_ENTRY( 0x0066, NtOpenDirectoryObject, 12 ) \
+    SYSCALL_ENTRY( 0x0067, NtOpenEvent, 12 ) \
+    SYSCALL_ENTRY( 0x0068, NtOpenFile, 24 ) \
+    SYSCALL_ENTRY( 0x0069, NtOpenIoCompletion, 12 ) \
+    SYSCALL_ENTRY( 0x006a, NtOpenJobObject, 12 ) \
+    SYSCALL_ENTRY( 0x006b, NtOpenKey, 12 ) \
+    SYSCALL_ENTRY( 0x006c, NtOpenKeyEx, 16 ) \
+    SYSCALL_ENTRY( 0x006d, NtOpenKeyTransacted, 16 ) \
+    SYSCALL_ENTRY( 0x006e, NtOpenKeyTransactedEx, 20 ) \
+    SYSCALL_ENTRY( 0x006f, NtOpenKeyedEvent, 12 ) \
+    SYSCALL_ENTRY( 0x0070, NtOpenMutant, 12 ) \
+    SYSCALL_ENTRY( 0x0071, NtOpenProcess, 16 ) \
+    SYSCALL_ENTRY( 0x0072, NtOpenProcessToken, 12 ) \
+    SYSCALL_ENTRY( 0x0073, NtOpenProcessTokenEx, 16 ) \
+    SYSCALL_ENTRY( 0x0074, NtOpenSection, 12 ) \
+    SYSCALL_ENTRY( 0x0075, NtOpenSemaphore, 12 ) \
+    SYSCALL_ENTRY( 0x0076, NtOpenSymbolicLinkObject, 12 ) \
+    SYSCALL_ENTRY( 0x0077, NtOpenThread, 16 ) \
+    SYSCALL_ENTRY( 0x0078, NtOpenThreadToken, 16 ) \
+    SYSCALL_ENTRY( 0x0079, NtOpenThreadTokenEx, 20 ) \
+    SYSCALL_ENTRY( 0x007a, NtOpenTimer, 12 ) \
+    SYSCALL_ENTRY( 0x007b, NtPowerInformation, 20 ) \
+    SYSCALL_ENTRY( 0x007c, NtPrivilegeCheck, 12 ) \
+    SYSCALL_ENTRY( 0x007d, NtProtectVirtualMemory, 20 ) \
+    SYSCALL_ENTRY( 0x007e, NtPulseEvent, 8 ) \
+    SYSCALL_ENTRY( 0x007f, NtQueryAttributesFile, 8 ) \
+    SYSCALL_ENTRY( 0x0080, NtQueryDefaultLocale, 8 ) \
+    SYSCALL_ENTRY( 0x0081, NtQueryDefaultUILanguage, 4 ) \
+    SYSCALL_ENTRY( 0x0082, NtQueryDirectoryFile, 44 ) \
+    SYSCALL_ENTRY( 0x0083, NtQueryDirectoryObject, 28 ) \
+    SYSCALL_ENTRY( 0x0084, NtQueryEaFile, 36 ) \
+    SYSCALL_ENTRY( 0x0085, NtQueryEvent, 20 ) \
+    SYSCALL_ENTRY( 0x0086, NtQueryFullAttributesFile, 8 ) \
+    SYSCALL_ENTRY( 0x0087, NtQueryInformationAtom, 20 ) \
+    SYSCALL_ENTRY( 0x0088, NtQueryInformationFile, 20 ) \
+    SYSCALL_ENTRY( 0x0089, NtQueryInformationJobObject, 20 ) \
+    SYSCALL_ENTRY( 0x008a, NtQueryInformationProcess, 20 ) \
+    SYSCALL_ENTRY( 0x008b, NtQueryInformationThread, 20 ) \
+    SYSCALL_ENTRY( 0x008c, NtQueryInformationToken, 20 ) \
+    SYSCALL_ENTRY( 0x008d, NtQueryInstallUILanguage, 4 ) \
+    SYSCALL_ENTRY( 0x008e, NtQueryIoCompletion, 20 ) \
+    SYSCALL_ENTRY( 0x008f, NtQueryKey, 20 ) \
+    SYSCALL_ENTRY( 0x0090, NtQueryLicenseValue, 20 ) \
+    SYSCALL_ENTRY( 0x0091, NtQueryMultipleValueKey, 24 ) \
+    SYSCALL_ENTRY( 0x0092, NtQueryMutant, 20 ) \
+    SYSCALL_ENTRY( 0x0093, NtQueryObject, 20 ) \
+    SYSCALL_ENTRY( 0x0094, NtQueryPerformanceCounter, 8 ) \
+    SYSCALL_ENTRY( 0x0095, NtQuerySection, 20 ) \
+    SYSCALL_ENTRY( 0x0096, NtQuerySecurityObject, 20 ) \
+    SYSCALL_ENTRY( 0x0097, NtQuerySemaphore, 20 ) \
+    SYSCALL_ENTRY( 0x0098, NtQuerySymbolicLinkObject, 12 ) \
+    SYSCALL_ENTRY( 0x0099, NtQuerySystemEnvironmentValue, 16 ) \
+    SYSCALL_ENTRY( 0x009a, NtQuerySystemEnvironmentValueEx, 20 ) \
+    SYSCALL_ENTRY( 0x009b, NtQuerySystemInformation, 16 ) \
+    SYSCALL_ENTRY( 0x009c, NtQuerySystemInformationEx, 24 ) \
+    SYSCALL_ENTRY( 0x009d, NtQuerySystemTime, 4 ) \
+    SYSCALL_ENTRY( 0x009e, NtQueryTimer, 20 ) \
+    SYSCALL_ENTRY( 0x009f, NtQueryTimerResolution, 12 ) \
+    SYSCALL_ENTRY( 0x00a0, NtQueryValueKey, 24 ) \
+    SYSCALL_ENTRY( 0x00a1, NtQueryVirtualMemory, 24 ) \
+    SYSCALL_ENTRY( 0x00a2, NtQueryVolumeInformationFile, 20 ) \
+    SYSCALL_ENTRY( 0x00a3, NtQueueApcThread, 20 ) \
+    SYSCALL_ENTRY( 0x00a4, NtQueueApcThreadEx, 24 ) \
+    SYSCALL_ENTRY( 0x00a5, NtRaiseException, 12 ) \
+    SYSCALL_ENTRY( 0x00a6, NtRaiseHardError, 24 ) \
+    SYSCALL_ENTRY( 0x00a7, NtReadFile, 36 ) \
+    SYSCALL_ENTRY( 0x00a8, NtReadFileScatter, 36 ) \
+    SYSCALL_ENTRY( 0x00a9, NtReadVirtualMemory, 20 ) \
+    SYSCALL_ENTRY( 0x00aa, NtRegisterThreadTerminatePort, 4 ) \
+    SYSCALL_ENTRY( 0x00ab, NtReleaseKeyedEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00ac, NtReleaseMutant, 8 ) \
+    SYSCALL_ENTRY( 0x00ad, NtReleaseSemaphore, 12 ) \
+    SYSCALL_ENTRY( 0x00ae, NtRemoveIoCompletion, 20 ) \
+    SYSCALL_ENTRY( 0x00af, NtRemoveIoCompletionEx, 24 ) \
+    SYSCALL_ENTRY( 0x00b0, NtRemoveProcessDebug, 8 ) \
+    SYSCALL_ENTRY( 0x00b1, NtRenameKey, 8 ) \
+    SYSCALL_ENTRY( 0x00b2, NtReplaceKey, 12 ) \
+    SYSCALL_ENTRY( 0x00b3, NtReplyWaitReceivePort, 16 ) \
+    SYSCALL_ENTRY( 0x00b4, NtRequestWaitReplyPort, 12 ) \
+    SYSCALL_ENTRY( 0x00b5, NtResetEvent, 8 ) \
+    SYSCALL_ENTRY( 0x00b6, NtResetWriteWatch, 12 ) \
+    SYSCALL_ENTRY( 0x00b7, NtRestoreKey, 12 ) \
+    SYSCALL_ENTRY( 0x00b8, NtResumeProcess, 4 ) \
+    SYSCALL_ENTRY( 0x00b9, NtResumeThread, 8 ) \
+    SYSCALL_ENTRY( 0x00ba, NtRollbackTransaction, 8 ) \
+    SYSCALL_ENTRY( 0x00bb, NtSaveKey, 8 ) \
+    SYSCALL_ENTRY( 0x00bc, NtSecureConnectPort, 36 ) \
+    SYSCALL_ENTRY( 0x00bd, NtSetContextThread, 8 ) \
+    SYSCALL_ENTRY( 0x00be, NtSetDebugFilterState, 12 ) \
+    SYSCALL_ENTRY( 0x00bf, NtSetDefaultLocale, 8 ) \
+    SYSCALL_ENTRY( 0x00c0, NtSetDefaultUILanguage, 4 ) \
+    SYSCALL_ENTRY( 0x00c1, NtSetEaFile, 16 ) \
+    SYSCALL_ENTRY( 0x00c2, NtSetEvent, 8 ) \
+    SYSCALL_ENTRY( 0x00c3, NtSetInformationDebugObject, 20 ) \
+    SYSCALL_ENTRY( 0x00c4, NtSetInformationFile, 20 ) \
+    SYSCALL_ENTRY( 0x00c5, NtSetInformationJobObject, 16 ) \
+    SYSCALL_ENTRY( 0x00c6, NtSetInformationKey, 16 ) \
+    SYSCALL_ENTRY( 0x00c7, NtSetInformationObject, 16 ) \
+    SYSCALL_ENTRY( 0x00c8, NtSetInformationProcess, 16 ) \
+    SYSCALL_ENTRY( 0x00c9, NtSetInformationThread, 16 ) \
+    SYSCALL_ENTRY( 0x00ca, NtSetInformationToken, 16 ) \
+    SYSCALL_ENTRY( 0x00cb, NtSetInformationVirtualMemory, 24 ) \
+    SYSCALL_ENTRY( 0x00cc, NtSetIntervalProfile, 8 ) \
+    SYSCALL_ENTRY( 0x00cd, NtSetIoCompletion, 20 ) \
+    SYSCALL_ENTRY( 0x00ce, NtSetIoCompletionEx, 24 ) \
+    SYSCALL_ENTRY( 0x00cf, NtSetLdtEntries, 24 ) \
+    SYSCALL_ENTRY( 0x00d0, NtSetSecurityObject, 12 ) \
+    SYSCALL_ENTRY( 0x00d1, NtSetSystemInformation, 12 ) \
+    SYSCALL_ENTRY( 0x00d2, NtSetSystemTime, 8 ) \
+    SYSCALL_ENTRY( 0x00d3, NtSetThreadExecutionState, 8 ) \
+    SYSCALL_ENTRY( 0x00d4, NtSetTimer, 28 ) \
+    SYSCALL_ENTRY( 0x00d5, NtSetTimerResolution, 12 ) \
+    SYSCALL_ENTRY( 0x00d6, NtSetValueKey, 24 ) \
+    SYSCALL_ENTRY( 0x00d7, NtSetVolumeInformationFile, 20 ) \
+    SYSCALL_ENTRY( 0x00d8, NtShutdownSystem, 4 ) \
+    SYSCALL_ENTRY( 0x00d9, NtSignalAndWaitForSingleObject, 16 ) \
+    SYSCALL_ENTRY( 0x00da, NtSuspendProcess, 4 ) \
+    SYSCALL_ENTRY( 0x00db, NtSuspendThread, 8 ) \
+    SYSCALL_ENTRY( 0x00dc, NtSystemDebugControl, 24 ) \
+    SYSCALL_ENTRY( 0x00dd, NtTerminateJobObject, 8 ) \
+    SYSCALL_ENTRY( 0x00de, NtTerminateProcess, 8 ) \
+    SYSCALL_ENTRY( 0x00df, NtTerminateThread, 8 ) \
+    SYSCALL_ENTRY( 0x00e0, NtTestAlert, 0 ) \
+    SYSCALL_ENTRY( 0x00e1, NtTraceControl, 24 ) \
+    SYSCALL_ENTRY( 0x00e2, NtUnloadDriver, 4 ) \
+    SYSCALL_ENTRY( 0x00e3, NtUnloadKey, 4 ) \
+    SYSCALL_ENTRY( 0x00e4, NtUnlockFile, 20 ) \
+    SYSCALL_ENTRY( 0x00e5, NtUnlockVirtualMemory, 16 ) \
+    SYSCALL_ENTRY( 0x00e6, NtUnmapViewOfSection, 8 ) \
+    SYSCALL_ENTRY( 0x00e7, NtUnmapViewOfSectionEx, 12 ) \
+    SYSCALL_ENTRY( 0x00e8, NtWaitForAlertByThreadId, 8 ) \
+    SYSCALL_ENTRY( 0x00e9, NtWaitForDebugEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00ea, NtWaitForKeyedEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00eb, NtWaitForMultipleObjects, 20 ) \
+    SYSCALL_ENTRY( 0x00ec, NtWaitForSingleObject, 12 ) \
+    SYSCALL_ENTRY( 0x00ed, NtWow64AllocateVirtualMemory64, 28 ) \
+    SYSCALL_ENTRY( 0x00ee, NtWow64GetNativeSystemInformation, 16 ) \
+    SYSCALL_ENTRY( 0x00ef, NtWow64IsProcessorFeaturePresent, 4 ) \
+    SYSCALL_ENTRY( 0x00f0, NtWow64QueryInformationProcess64, 20 ) \
+    SYSCALL_ENTRY( 0x00f1, NtWow64ReadVirtualMemory64, 28 ) \
+    SYSCALL_ENTRY( 0x00f2, NtWow64WriteVirtualMemory64, 28 ) \
+    SYSCALL_ENTRY( 0x00f3, NtWriteFile, 36 ) \
+    SYSCALL_ENTRY( 0x00f4, NtWriteFileGather, 36 ) \
+    SYSCALL_ENTRY( 0x00f5, NtWriteVirtualMemory, 20 ) \
+    SYSCALL_ENTRY( 0x00f6, NtYieldExecution, 0 ) \
+    SYSCALL_ENTRY( 0x00f7, __wine_dbg_ftrace, 12 ) \
+    SYSCALL_ENTRY( 0x00f8, __wine_needs_override_large_address_aware, 0 ) \
+    SYSCALL_ENTRY( 0x00f9, __wine_set_unix_env, 8 ) \
+    SYSCALL_ENTRY( 0x00fa, wine_nt_to_unix_file_name, 16 ) \
+    SYSCALL_ENTRY( 0x00fb, wine_unix_to_nt_file_name, 12 )
+
+#define ALL_SYSCALLS64 \
+    SYSCALL_ENTRY( 0x0000, NtAcceptConnectPort, 48 ) \
+    SYSCALL_ENTRY( 0x0001, NtAccessCheck, 64 ) \
+    SYSCALL_ENTRY( 0x0002, NtAccessCheckAndAuditAlarm, 88 ) \
+    SYSCALL_ENTRY( 0x0003, NtAddAtom, 24 ) \
+    SYSCALL_ENTRY( 0x0004, NtAdjustGroupsToken, 48 ) \
+    SYSCALL_ENTRY( 0x0005, NtAdjustPrivilegesToken, 48 ) \
+    SYSCALL_ENTRY( 0x0006, NtAlertResumeThread, 16 ) \
+    SYSCALL_ENTRY( 0x0007, NtAlertThread, 8 ) \
+    SYSCALL_ENTRY( 0x0008, NtAlertThreadByThreadId, 8 ) \
+    SYSCALL_ENTRY( 0x0009, NtAllocateLocallyUniqueId, 8 ) \
+    SYSCALL_ENTRY( 0x000a, NtAllocateReserveObject, 24 ) \
+    SYSCALL_ENTRY( 0x000b, NtAllocateUuids, 32 ) \
+    SYSCALL_ENTRY( 0x000c, NtAllocateVirtualMemory, 48 ) \
+    SYSCALL_ENTRY( 0x000d, NtAllocateVirtualMemoryEx, 56 ) \
+    SYSCALL_ENTRY( 0x000e, NtAreMappedFilesTheSame, 16 ) \
+    SYSCALL_ENTRY( 0x000f, NtAssignProcessToJobObject, 16 ) \
+    SYSCALL_ENTRY( 0x0010, NtCallbackReturn, 24 ) \
+    SYSCALL_ENTRY( 0x0011, NtCancelIoFile, 16 ) \
+    SYSCALL_ENTRY( 0x0012, NtCancelIoFileEx, 24 ) \
+    SYSCALL_ENTRY( 0x0013, NtCancelSynchronousIoFile, 24 ) \
+    SYSCALL_ENTRY( 0x0014, NtCancelTimer, 16 ) \
+    SYSCALL_ENTRY( 0x0015, NtClearEvent, 8 ) \
+    SYSCALL_ENTRY( 0x0016, NtClose, 8 ) \
+    SYSCALL_ENTRY( 0x0017, NtCommitTransaction, 16 ) \
+    SYSCALL_ENTRY( 0x0018, NtCompareObjects, 16 ) \
+    SYSCALL_ENTRY( 0x0019, NtCompareTokens, 24 ) \
+    SYSCALL_ENTRY( 0x001a, NtCompleteConnectPort, 8 ) \
+    SYSCALL_ENTRY( 0x001b, NtConnectPort, 64 ) \
+    SYSCALL_ENTRY( 0x001c, NtContinue, 16 ) \
+    SYSCALL_ENTRY( 0x001d, NtContinueEx, 16 ) \
+    SYSCALL_ENTRY( 0x001e, NtConvertBetweenAuxiliaryCounterAndPerformanceCounter, 32 ) \
+    SYSCALL_ENTRY( 0x001f, NtCreateDebugObject, 32 ) \
+    SYSCALL_ENTRY( 0x0020, NtCreateDirectoryObject, 24 ) \
+    SYSCALL_ENTRY( 0x0021, NtCreateEvent, 40 ) \
+    SYSCALL_ENTRY( 0x0022, NtCreateFile, 88 ) \
+    SYSCALL_ENTRY( 0x0023, NtCreateIoCompletion, 32 ) \
+    SYSCALL_ENTRY( 0x0024, NtCreateJobObject, 24 ) \
+    SYSCALL_ENTRY( 0x0025, NtCreateKey, 56 ) \
+    SYSCALL_ENTRY( 0x0026, NtCreateKeyTransacted, 64 ) \
+    SYSCALL_ENTRY( 0x0027, NtCreateKeyedEvent, 32 ) \
+    SYSCALL_ENTRY( 0x0028, NtCreateLowBoxToken, 72 ) \
+    SYSCALL_ENTRY( 0x0029, NtCreateMailslotFile, 64 ) \
+    SYSCALL_ENTRY( 0x002a, NtCreateMutant, 32 ) \
+    SYSCALL_ENTRY( 0x002b, NtCreateNamedPipeFile, 112 ) \
+    SYSCALL_ENTRY( 0x002c, NtCreatePagingFile, 32 ) \
+    SYSCALL_ENTRY( 0x002d, NtCreatePort, 40 ) \
+    SYSCALL_ENTRY( 0x002e, NtCreateSection, 56 ) \
+    SYSCALL_ENTRY( 0x002f, NtCreateSectionEx, 72 ) \
+    SYSCALL_ENTRY( 0x0030, NtCreateSemaphore, 40 ) \
+    SYSCALL_ENTRY( 0x0031, NtCreateSymbolicLinkObject, 32 ) \
+    SYSCALL_ENTRY( 0x0032, NtCreateThread, 64 ) \
+    SYSCALL_ENTRY( 0x0033, NtCreateThreadEx, 88 ) \
+    SYSCALL_ENTRY( 0x0034, NtCreateTimer, 32 ) \
+    SYSCALL_ENTRY( 0x0035, NtCreateToken, 104 ) \
+    SYSCALL_ENTRY( 0x0036, NtCreateTransaction, 80 ) \
+    SYSCALL_ENTRY( 0x0037, NtCreateUserProcess, 88 ) \
+    SYSCALL_ENTRY( 0x0038, NtDebugActiveProcess, 16 ) \
+    SYSCALL_ENTRY( 0x0039, NtDebugContinue, 24 ) \
+    SYSCALL_ENTRY( 0x003a, NtDelayExecution, 16 ) \
+    SYSCALL_ENTRY( 0x003b, NtDeleteAtom, 8 ) \
+    SYSCALL_ENTRY( 0x003c, NtDeleteFile, 8 ) \
+    SYSCALL_ENTRY( 0x003d, NtDeleteKey, 8 ) \
+    SYSCALL_ENTRY( 0x003e, NtDeleteValueKey, 16 ) \
+    SYSCALL_ENTRY( 0x003f, NtDeviceIoControlFile, 80 ) \
+    SYSCALL_ENTRY( 0x0040, NtDisplayString, 8 ) \
+    SYSCALL_ENTRY( 0x0041, NtDuplicateObject, 56 ) \
+    SYSCALL_ENTRY( 0x0042, NtDuplicateToken, 48 ) \
+    SYSCALL_ENTRY( 0x0043, NtEnumerateKey, 48 ) \
+    SYSCALL_ENTRY( 0x0044, NtEnumerateValueKey, 48 ) \
+    SYSCALL_ENTRY( 0x0045, NtFilterToken, 48 ) \
+    SYSCALL_ENTRY( 0x0046, NtFindAtom, 24 ) \
+    SYSCALL_ENTRY( 0x0047, NtFlushBuffersFile, 16 ) \
+    SYSCALL_ENTRY( 0x0048, NtFlushBuffersFileEx, 40 ) \
+    SYSCALL_ENTRY( 0x0049, NtFlushInstructionCache, 24 ) \
+    SYSCALL_ENTRY( 0x004a, NtFlushKey, 8 ) \
+    SYSCALL_ENTRY( 0x004b, NtFlushProcessWriteBuffers, 0 ) \
+    SYSCALL_ENTRY( 0x004c, NtFlushVirtualMemory, 32 ) \
+    SYSCALL_ENTRY( 0x004d, NtFreeVirtualMemory, 32 ) \
+    SYSCALL_ENTRY( 0x004e, NtFsControlFile, 80 ) \
+    SYSCALL_ENTRY( 0x004f, NtGetContextThread, 16 ) \
+    SYSCALL_ENTRY( 0x0050, NtGetCurrentProcessorNumber, 0 ) \
+    SYSCALL_ENTRY( 0x0051, NtGetNextThread, 48 ) \
+    SYSCALL_ENTRY( 0x0052, NtGetNlsSectionPtr, 40 ) \
+    SYSCALL_ENTRY( 0x0053, NtGetWriteWatch, 56 ) \
+    SYSCALL_ENTRY( 0x0054, NtImpersonateAnonymousToken, 8 ) \
+    SYSCALL_ENTRY( 0x0055, NtInitializeNlsFiles, 24 ) \
+    SYSCALL_ENTRY( 0x0056, NtInitiatePowerAction, 32 ) \
+    SYSCALL_ENTRY( 0x0057, NtIsProcessInJob, 16 ) \
+    SYSCALL_ENTRY( 0x0058, NtListenPort, 16 ) \
+    SYSCALL_ENTRY( 0x0059, NtLoadDriver, 8 ) \
+    SYSCALL_ENTRY( 0x005a, NtLoadKey, 16 ) \
+    SYSCALL_ENTRY( 0x005b, NtLoadKey2, 24 ) \
+    SYSCALL_ENTRY( 0x005c, NtLoadKeyEx, 64 ) \
+    SYSCALL_ENTRY( 0x005d, NtLockFile, 80 ) \
+    SYSCALL_ENTRY( 0x005e, NtLockVirtualMemory, 32 ) \
+    SYSCALL_ENTRY( 0x005f, NtMakePermanentObject, 8 ) \
+    SYSCALL_ENTRY( 0x0060, NtMakeTemporaryObject, 8 ) \
+    SYSCALL_ENTRY( 0x0061, NtMapViewOfSection, 80 ) \
+    SYSCALL_ENTRY( 0x0062, NtMapViewOfSectionEx, 72 ) \
+    SYSCALL_ENTRY( 0x0063, NtNotifyChangeDirectoryFile, 72 ) \
+    SYSCALL_ENTRY( 0x0064, NtNotifyChangeKey, 80 ) \
+    SYSCALL_ENTRY( 0x0065, NtNotifyChangeMultipleKeys, 96 ) \
+    SYSCALL_ENTRY( 0x0066, NtOpenDirectoryObject, 24 ) \
+    SYSCALL_ENTRY( 0x0067, NtOpenEvent, 24 ) \
+    SYSCALL_ENTRY( 0x0068, NtOpenFile, 48 ) \
+    SYSCALL_ENTRY( 0x0069, NtOpenIoCompletion, 24 ) \
+    SYSCALL_ENTRY( 0x006a, NtOpenJobObject, 24 ) \
+    SYSCALL_ENTRY( 0x006b, NtOpenKey, 24 ) \
+    SYSCALL_ENTRY( 0x006c, NtOpenKeyEx, 32 ) \
+    SYSCALL_ENTRY( 0x006d, NtOpenKeyTransacted, 32 ) \
+    SYSCALL_ENTRY( 0x006e, NtOpenKeyTransactedEx, 40 ) \
+    SYSCALL_ENTRY( 0x006f, NtOpenKeyedEvent, 24 ) \
+    SYSCALL_ENTRY( 0x0070, NtOpenMutant, 24 ) \
+    SYSCALL_ENTRY( 0x0071, NtOpenProcess, 32 ) \
+    SYSCALL_ENTRY( 0x0072, NtOpenProcessToken, 24 ) \
+    SYSCALL_ENTRY( 0x0073, NtOpenProcessTokenEx, 32 ) \
+    SYSCALL_ENTRY( 0x0074, NtOpenSection, 24 ) \
+    SYSCALL_ENTRY( 0x0075, NtOpenSemaphore, 24 ) \
+    SYSCALL_ENTRY( 0x0076, NtOpenSymbolicLinkObject, 24 ) \
+    SYSCALL_ENTRY( 0x0077, NtOpenThread, 32 ) \
+    SYSCALL_ENTRY( 0x0078, NtOpenThreadToken, 32 ) \
+    SYSCALL_ENTRY( 0x0079, NtOpenThreadTokenEx, 40 ) \
+    SYSCALL_ENTRY( 0x007a, NtOpenTimer, 24 ) \
+    SYSCALL_ENTRY( 0x007b, NtPowerInformation, 40 ) \
+    SYSCALL_ENTRY( 0x007c, NtPrivilegeCheck, 24 ) \
+    SYSCALL_ENTRY( 0x007d, NtProtectVirtualMemory, 40 ) \
+    SYSCALL_ENTRY( 0x007e, NtPulseEvent, 16 ) \
+    SYSCALL_ENTRY( 0x007f, NtQueryAttributesFile, 16 ) \
+    SYSCALL_ENTRY( 0x0080, NtQueryDefaultLocale, 16 ) \
+    SYSCALL_ENTRY( 0x0081, NtQueryDefaultUILanguage, 8 ) \
+    SYSCALL_ENTRY( 0x0082, NtQueryDirectoryFile, 88 ) \
+    SYSCALL_ENTRY( 0x0083, NtQueryDirectoryObject, 56 ) \
+    SYSCALL_ENTRY( 0x0084, NtQueryEaFile, 72 ) \
+    SYSCALL_ENTRY( 0x0085, NtQueryEvent, 40 ) \
+    SYSCALL_ENTRY( 0x0086, NtQueryFullAttributesFile, 16 ) \
+    SYSCALL_ENTRY( 0x0087, NtQueryInformationAtom, 40 ) \
+    SYSCALL_ENTRY( 0x0088, NtQueryInformationFile, 40 ) \
+    SYSCALL_ENTRY( 0x0089, NtQueryInformationJobObject, 40 ) \
+    SYSCALL_ENTRY( 0x008a, NtQueryInformationProcess, 40 ) \
+    SYSCALL_ENTRY( 0x008b, NtQueryInformationThread, 40 ) \
+    SYSCALL_ENTRY( 0x008c, NtQueryInformationToken, 40 ) \
+    SYSCALL_ENTRY( 0x008d, NtQueryInstallUILanguage, 8 ) \
+    SYSCALL_ENTRY( 0x008e, NtQueryIoCompletion, 40 ) \
+    SYSCALL_ENTRY( 0x008f, NtQueryKey, 40 ) \
+    SYSCALL_ENTRY( 0x0090, NtQueryLicenseValue, 40 ) \
+    SYSCALL_ENTRY( 0x0091, NtQueryMultipleValueKey, 48 ) \
+    SYSCALL_ENTRY( 0x0092, NtQueryMutant, 40 ) \
+    SYSCALL_ENTRY( 0x0093, NtQueryObject, 40 ) \
+    SYSCALL_ENTRY( 0x0094, NtQueryPerformanceCounter, 16 ) \
+    SYSCALL_ENTRY( 0x0095, NtQuerySection, 40 ) \
+    SYSCALL_ENTRY( 0x0096, NtQuerySecurityObject, 40 ) \
+    SYSCALL_ENTRY( 0x0097, NtQuerySemaphore, 40 ) \
+    SYSCALL_ENTRY( 0x0098, NtQuerySymbolicLinkObject, 24 ) \
+    SYSCALL_ENTRY( 0x0099, NtQuerySystemEnvironmentValue, 32 ) \
+    SYSCALL_ENTRY( 0x009a, NtQuerySystemEnvironmentValueEx, 40 ) \
+    SYSCALL_ENTRY( 0x009b, NtQuerySystemInformation, 32 ) \
+    SYSCALL_ENTRY( 0x009c, NtQuerySystemInformationEx, 48 ) \
+    SYSCALL_ENTRY( 0x009d, NtQuerySystemTime, 8 ) \
+    SYSCALL_ENTRY( 0x009e, NtQueryTimer, 40 ) \
+    SYSCALL_ENTRY( 0x009f, NtQueryTimerResolution, 24 ) \
+    SYSCALL_ENTRY( 0x00a0, NtQueryValueKey, 48 ) \
+    SYSCALL_ENTRY( 0x00a1, NtQueryVirtualMemory, 48 ) \
+    SYSCALL_ENTRY( 0x00a2, NtQueryVolumeInformationFile, 40 ) \
+    SYSCALL_ENTRY( 0x00a3, NtQueueApcThread, 40 ) \
+    SYSCALL_ENTRY( 0x00a4, NtQueueApcThreadEx, 48 ) \
+    SYSCALL_ENTRY( 0x00a5, NtRaiseException, 24 ) \
+    SYSCALL_ENTRY( 0x00a6, NtRaiseHardError, 48 ) \
+    SYSCALL_ENTRY( 0x00a7, NtReadFile, 72 ) \
+    SYSCALL_ENTRY( 0x00a8, NtReadFileScatter, 72 ) \
+    SYSCALL_ENTRY( 0x00a9, NtReadVirtualMemory, 40 ) \
+    SYSCALL_ENTRY( 0x00aa, NtRegisterThreadTerminatePort, 8 ) \
+    SYSCALL_ENTRY( 0x00ab, NtReleaseKeyedEvent, 32 ) \
+    SYSCALL_ENTRY( 0x00ac, NtReleaseMutant, 16 ) \
+    SYSCALL_ENTRY( 0x00ad, NtReleaseSemaphore, 24 ) \
+    SYSCALL_ENTRY( 0x00ae, NtRemoveIoCompletion, 40 ) \
+    SYSCALL_ENTRY( 0x00af, NtRemoveIoCompletionEx, 48 ) \
+    SYSCALL_ENTRY( 0x00b0, NtRemoveProcessDebug, 16 ) \
+    SYSCALL_ENTRY( 0x00b1, NtRenameKey, 16 ) \
+    SYSCALL_ENTRY( 0x00b2, NtReplaceKey, 24 ) \
+    SYSCALL_ENTRY( 0x00b3, NtReplyWaitReceivePort, 32 ) \
+    SYSCALL_ENTRY( 0x00b4, NtRequestWaitReplyPort, 24 ) \
+    SYSCALL_ENTRY( 0x00b5, NtResetEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00b6, NtResetWriteWatch, 24 ) \
+    SYSCALL_ENTRY( 0x00b7, NtRestoreKey, 24 ) \
+    SYSCALL_ENTRY( 0x00b8, NtResumeProcess, 8 ) \
+    SYSCALL_ENTRY( 0x00b9, NtResumeThread, 16 ) \
+    SYSCALL_ENTRY( 0x00ba, NtRollbackTransaction, 16 ) \
+    SYSCALL_ENTRY( 0x00bb, NtSaveKey, 16 ) \
+    SYSCALL_ENTRY( 0x00bc, NtSecureConnectPort, 72 ) \
+    SYSCALL_ENTRY( 0x00bd, NtSetContextThread, 16 ) \
+    SYSCALL_ENTRY( 0x00be, NtSetDebugFilterState, 24 ) \
+    SYSCALL_ENTRY( 0x00bf, NtSetDefaultLocale, 16 ) \
+    SYSCALL_ENTRY( 0x00c0, NtSetDefaultUILanguage, 8 ) \
+    SYSCALL_ENTRY( 0x00c1, NtSetEaFile, 32 ) \
+    SYSCALL_ENTRY( 0x00c2, NtSetEvent, 16 ) \
+    SYSCALL_ENTRY( 0x00c3, NtSetInformationDebugObject, 40 ) \
+    SYSCALL_ENTRY( 0x00c4, NtSetInformationFile, 40 ) \
+    SYSCALL_ENTRY( 0x00c5, NtSetInformationJobObject, 32 ) \
+    SYSCALL_ENTRY( 0x00c6, NtSetInformationKey, 32 ) \
+    SYSCALL_ENTRY( 0x00c7, NtSetInformationObject, 32 ) \
+    SYSCALL_ENTRY( 0x00c8, NtSetInformationProcess, 32 ) \
+    SYSCALL_ENTRY( 0x00c9, NtSetInformationThread, 32 ) \
+    SYSCALL_ENTRY( 0x00ca, NtSetInformationToken, 32 ) \
+    SYSCALL_ENTRY( 0x00cb, NtSetInformationVirtualMemory, 48 ) \
+    SYSCALL_ENTRY( 0x00cc, NtSetIntervalProfile, 16 ) \
+    SYSCALL_ENTRY( 0x00cd, NtSetIoCompletion, 40 ) \
+    SYSCALL_ENTRY( 0x00ce, NtSetIoCompletionEx, 48 ) \
+    SYSCALL_ENTRY( 0x00cf, NtSetLdtEntries, 32 ) \
+    SYSCALL_ENTRY( 0x00d0, NtSetSecurityObject, 24 ) \
+    SYSCALL_ENTRY( 0x00d1, NtSetSystemInformation, 24 ) \
+    SYSCALL_ENTRY( 0x00d2, NtSetSystemTime, 16 ) \
+    SYSCALL_ENTRY( 0x00d3, NtSetThreadExecutionState, 16 ) \
+    SYSCALL_ENTRY( 0x00d4, NtSetTimer, 56 ) \
+    SYSCALL_ENTRY( 0x00d5, NtSetTimerResolution, 24 ) \
+    SYSCALL_ENTRY( 0x00d6, NtSetValueKey, 48 ) \
+    SYSCALL_ENTRY( 0x00d7, NtSetVolumeInformationFile, 40 ) \
+    SYSCALL_ENTRY( 0x00d8, NtShutdownSystem, 8 ) \
+    SYSCALL_ENTRY( 0x00d9, NtSignalAndWaitForSingleObject, 32 ) \
+    SYSCALL_ENTRY( 0x00da, NtSuspendProcess, 8 ) \
+    SYSCALL_ENTRY( 0x00db, NtSuspendThread, 16 ) \
+    SYSCALL_ENTRY( 0x00dc, NtSystemDebugControl, 48 ) \
+    SYSCALL_ENTRY( 0x00dd, NtTerminateJobObject, 16 ) \
+    SYSCALL_ENTRY( 0x00de, NtTerminateProcess, 16 ) \
+    SYSCALL_ENTRY( 0x00df, NtTerminateThread, 16 ) \
+    SYSCALL_ENTRY( 0x00e0, NtTestAlert, 0 ) \
+    SYSCALL_ENTRY( 0x00e1, NtTraceControl, 48 ) \
+    SYSCALL_ENTRY( 0x00e2, NtUnloadDriver, 8 ) \
+    SYSCALL_ENTRY( 0x00e3, NtUnloadKey, 8 ) \
+    SYSCALL_ENTRY( 0x00e4, NtUnlockFile, 40 ) \
+    SYSCALL_ENTRY( 0x00e5, NtUnlockVirtualMemory, 32 ) \
+    SYSCALL_ENTRY( 0x00e6, NtUnmapViewOfSection, 16 ) \
+    SYSCALL_ENTRY( 0x00e7, NtUnmapViewOfSectionEx, 24 ) \
+    SYSCALL_ENTRY( 0x00e8, NtWaitForAlertByThreadId, 16 ) \
+    SYSCALL_ENTRY( 0x00e9, NtWaitForDebugEvent, 32 ) \
+    SYSCALL_ENTRY( 0x00ea, NtWaitForKeyedEvent, 32 ) \
+    SYSCALL_ENTRY( 0x00eb, NtWaitForMultipleObjects, 40 ) \
+    SYSCALL_ENTRY( 0x00ec, NtWaitForSingleObject, 24 ) \
+    SYSCALL_ENTRY( 0x00ed, NtWriteFile, 72 ) \
+    SYSCALL_ENTRY( 0x00ee, NtWriteFileGather, 72 ) \
+    SYSCALL_ENTRY( 0x00ef, NtWriteVirtualMemory, 40 ) \
+    SYSCALL_ENTRY( 0x00f0, NtYieldExecution, 0 ) \
+    SYSCALL_ENTRY( 0x00f1, __wine_dbg_ftrace, 24 ) \
+    SYSCALL_ENTRY( 0x00f2, __wine_needs_override_large_address_aware, 0 ) \
+    SYSCALL_ENTRY( 0x00f3, __wine_set_unix_env, 16 ) \
+    SYSCALL_ENTRY( 0x00f4, wine_nt_to_unix_file_name, 32 ) \
+    SYSCALL_ENTRY( 0x00f5, wine_unix_to_nt_file_name, 24 )
diff --git a/dlls/user32/user32.spec b/dlls/user32/user32.spec
index 77e99904014..82c244aa1b1 100644
--- a/dlls/user32/user32.spec
+++ b/dlls/user32/user32.spec
@@ -163,19 +163,19 @@
 2661 stub @
 # 2662 stub RIMRegisterForInputEx
 # 2663 stub RIMOnAsyncPnpWorkNotification
-# 2664 stub ShellMigrateWindow
-# 2665 stub SetAdditionalForegroundBoostProcesses
-# 2666 stub RegisterForTooltipDismissNotification
-# 2667 stub RegisterForCustomDockTargets
-# 2668 stub GetClipboardMetadata
-2669 stub @
-2670 stub @
-# 2671 stub ShellRegisterHotKey
-# 2672 stub SetUserObjectCapability
-# 2673 stub SetWindowMessageCapability
-# 2674 stub ShellForegroundBoostProcess
-# 2675 stub SuppressWindowActions
-# 2676 stub GetSuppressedWindowActions
+2664 stub ShellMigrateWindow  # NtUserShellMigrateWindow
+2665 stdcall SetAdditionalForegroundBoostProcesses(ptr long ptr) NtUserSetAdditionalForegroundBoostProcesses
+2666 stub RegisterForTooltipDismissNotification  # NtUserRegisterForTooltipDismissNotification
+2667 stub RegisterForCustomDockTargets  # NtUserRegisterForCustomDockTargets
+2668 stub GetClipboardMetadata  # NtUserGetClipboardMetadata
+2669 stub -noname InitThreadCoreMessagingIocp2  # NtUserInitThreadCoreMessagingIocp2
+2670 stub -noname DrainThreadCoreMessagingCompletions2  # NtUserDrainThreadCoreMessagingCompletions2
+2671 stub ShellRegisterHotKey  # NtUserShellRegisterHotKey
+2672 stub SetUserObjectCapability  # NtUserSetUserObjectCapability
+2673 stub SetWindowMessageCapability  # NtUserSetWindowMessageCapability
+2674 stub ShellForegroundBoostProcess  # NtUserShellForegroundBoostProcess
+2675 stub SuppressWindowActions  # NtUserSuppressWindowActions
+2676 stub GetSuppressedWindowActions  # NtUserGetSuppressedWindowActions
 
 # 2680 stub DwmWindowNotificationsEnabled
 # 2681 stub ApplyWindowAction
diff --git a/dlls/win32u/main.c b/dlls/win32u/main.c
index 4e81760c9d2..22140454fd3 100644
--- a/dlls/win32u/main.c
+++ b/dlls/win32u/main.c
@@ -2025,6 +2025,11 @@ HWND SYSCALL_API NtUserSetActiveWindow( HWND hwnd )
     SYSCALL_FUNC( NtUserSetActiveWindow );
 }
 
+BOOL SYSCALL_API NtUserSetAdditionalForegroundBoostProcesses( HWND hwnd, DWORD count, HANDLE *handles )
+{
+    SYSCALL_FUNC( NtUserSetAdditionalForegroundBoostProcesses );
+}
+
 HWND SYSCALL_API NtUserSetCapture( HWND hwnd )
 {
     SYSCALL_FUNC( NtUserSetCapture );
diff --git a/dlls/win32u/sysparams.c b/dlls/win32u/sysparams.c
index 3f7635c0f75..bbadae17b76 100644
--- a/dlls/win32u/sysparams.c
+++ b/dlls/win32u/sysparams.c
@@ -7166,6 +7166,16 @@ BOOL WINAPI NtUserMessageBeep( UINT type )
     return TRUE;
 }
 
+/***********************************************************************
+ *	     NtUserSetAdditionalForegroundBoostProcesses    (win32u.@)
+ */
+BOOL WINAPI NtUserSetAdditionalForegroundBoostProcesses( HWND hwnd, DWORD count, HANDLE *handles )
+{
+    FIXME( "%p %u %p stub!\n", hwnd, (unsigned int)count, handles );
+    RtlSetLastWin32Error( ERROR_CALL_NOT_IMPLEMENTED );
+    return FALSE;
+}
+
 static DWORD exiting_thread_id;
 
 /**********************************************************************
diff --git a/dlls/win32u/win32syscalls.h b/dlls/win32u/win32syscalls.h
new file mode 100644
index 00000000000..8a9e4b00376
--- /dev/null
+++ b/dlls/win32u/win32syscalls.h
@@ -0,0 +1,891 @@
+/* Automatically generated by tools/make_specfiles */
+
+#define ALL_SYSCALLS32 \
+    SYSCALL_ENTRY( 0x0000, NtGdiAbortDoc, 4 ) \
+    SYSCALL_ENTRY( 0x0001, NtGdiAbortPath, 4 ) \
+    SYSCALL_ENTRY( 0x0002, NtGdiAddFontMemResourceEx, 20 ) \
+    SYSCALL_ENTRY( 0x0003, NtGdiAddFontResourceW, 24 ) \
+    SYSCALL_ENTRY( 0x0004, NtGdiAlphaBlend, 48 ) \
+    SYSCALL_ENTRY( 0x0005, NtGdiAngleArc, 24 ) \
+    SYSCALL_ENTRY( 0x0006, NtGdiArcInternal, 40 ) \
+    SYSCALL_ENTRY( 0x0007, NtGdiBeginPath, 4 ) \
+    SYSCALL_ENTRY( 0x0008, NtGdiBitBlt, 44 ) \
+    SYSCALL_ENTRY( 0x0009, NtGdiCloseFigure, 4 ) \
+    SYSCALL_ENTRY( 0x000a, NtGdiCombineRgn, 16 ) \
+    SYSCALL_ENTRY( 0x000b, NtGdiComputeXformCoefficients, 4 ) \
+    SYSCALL_ENTRY( 0x000c, NtGdiCreateBitmap, 20 ) \
+    SYSCALL_ENTRY( 0x000d, NtGdiCreateClientObj, 4 ) \
+    SYSCALL_ENTRY( 0x000e, NtGdiCreateCompatibleBitmap, 12 ) \
+    SYSCALL_ENTRY( 0x000f, NtGdiCreateCompatibleDC, 4 ) \
+    SYSCALL_ENTRY( 0x0010, NtGdiCreateDIBBrush, 24 ) \
+    SYSCALL_ENTRY( 0x0011, NtGdiCreateDIBSection, 36 ) \
+    SYSCALL_ENTRY( 0x0012, NtGdiCreateDIBitmapInternal, 44 ) \
+    SYSCALL_ENTRY( 0x0013, NtGdiCreateEllipticRgn, 16 ) \
+    SYSCALL_ENTRY( 0x0014, NtGdiCreateHalftonePalette, 4 ) \
+    SYSCALL_ENTRY( 0x0015, NtGdiCreateHatchBrushInternal, 12 ) \
+    SYSCALL_ENTRY( 0x0016, NtGdiCreateMetafileDC, 4 ) \
+    SYSCALL_ENTRY( 0x0017, NtGdiCreatePaletteInternal, 8 ) \
+    SYSCALL_ENTRY( 0x0018, NtGdiCreatePatternBrushInternal, 12 ) \
+    SYSCALL_ENTRY( 0x0019, NtGdiCreatePen, 16 ) \
+    SYSCALL_ENTRY( 0x001a, NtGdiCreateRectRgn, 16 ) \
+    SYSCALL_ENTRY( 0x001b, NtGdiCreateRoundRectRgn, 24 ) \
+    SYSCALL_ENTRY( 0x001c, NtGdiCreateSolidBrush, 8 ) \
+    SYSCALL_ENTRY( 0x001d, NtGdiDdDDICheckVidPnExclusiveOwnership, 4 ) \
+    SYSCALL_ENTRY( 0x001e, NtGdiDdDDICloseAdapter, 4 ) \
+    SYSCALL_ENTRY( 0x001f, NtGdiDdDDICreateAllocation, 4 ) \
+    SYSCALL_ENTRY( 0x0020, NtGdiDdDDICreateAllocation2, 4 ) \
+    SYSCALL_ENTRY( 0x0021, NtGdiDdDDICreateDCFromMemory, 4 ) \
+    SYSCALL_ENTRY( 0x0022, NtGdiDdDDICreateDevice, 4 ) \
+    SYSCALL_ENTRY( 0x0023, NtGdiDdDDICreateKeyedMutex, 4 ) \
+    SYSCALL_ENTRY( 0x0024, NtGdiDdDDICreateKeyedMutex2, 4 ) \
+    SYSCALL_ENTRY( 0x0025, NtGdiDdDDICreateSynchronizationObject, 4 ) \
+    SYSCALL_ENTRY( 0x0026, NtGdiDdDDICreateSynchronizationObject2, 4 ) \
+    SYSCALL_ENTRY( 0x0027, NtGdiDdDDIDestroyAllocation, 4 ) \
+    SYSCALL_ENTRY( 0x0028, NtGdiDdDDIDestroyAllocation2, 4 ) \
+    SYSCALL_ENTRY( 0x0029, NtGdiDdDDIDestroyDCFromMemory, 4 ) \
+    SYSCALL_ENTRY( 0x002a, NtGdiDdDDIDestroyDevice, 4 ) \
+    SYSCALL_ENTRY( 0x002b, NtGdiDdDDIDestroyKeyedMutex, 4 ) \
+    SYSCALL_ENTRY( 0x002c, NtGdiDdDDIDestroySynchronizationObject, 4 ) \
+    SYSCALL_ENTRY( 0x002d, NtGdiDdDDIEnumAdapters, 4 ) \
+    SYSCALL_ENTRY( 0x002e, NtGdiDdDDIEnumAdapters2, 4 ) \
+    SYSCALL_ENTRY( 0x002f, NtGdiDdDDIEscape, 4 ) \
+    SYSCALL_ENTRY( 0x0030, NtGdiDdDDIOpenAdapterFromDeviceName, 4 ) \
+    SYSCALL_ENTRY( 0x0031, NtGdiDdDDIOpenAdapterFromHdc, 4 ) \
+    SYSCALL_ENTRY( 0x0032, NtGdiDdDDIOpenAdapterFromLuid, 4 ) \
+    SYSCALL_ENTRY( 0x0033, NtGdiDdDDIOpenKeyedMutex, 4 ) \
+    SYSCALL_ENTRY( 0x0034, NtGdiDdDDIOpenKeyedMutex2, 4 ) \
+    SYSCALL_ENTRY( 0x0035, NtGdiDdDDIOpenKeyedMutexFromNtHandle, 4 ) \
+    SYSCALL_ENTRY( 0x0036, NtGdiDdDDIOpenResource, 4 ) \
+    SYSCALL_ENTRY( 0x0037, NtGdiDdDDIOpenResource2, 4 ) \
+    SYSCALL_ENTRY( 0x0038, NtGdiDdDDIOpenResourceFromNtHandle, 4 ) \
+    SYSCALL_ENTRY( 0x0039, NtGdiDdDDIOpenSyncObjectFromNtHandle, 4 ) \
+    SYSCALL_ENTRY( 0x003a, NtGdiDdDDIOpenSyncObjectFromNtHandle2, 4 ) \
+    SYSCALL_ENTRY( 0x003b, NtGdiDdDDIOpenSyncObjectNtHandleFromName, 4 ) \
+    SYSCALL_ENTRY( 0x003c, NtGdiDdDDIOpenSynchronizationObject, 4 ) \
+    SYSCALL_ENTRY( 0x003d, NtGdiDdDDIQueryAdapterInfo, 4 ) \
+    SYSCALL_ENTRY( 0x003e, NtGdiDdDDIQueryResourceInfo, 4 ) \
+    SYSCALL_ENTRY( 0x003f, NtGdiDdDDIQueryResourceInfoFromNtHandle, 4 ) \
+    SYSCALL_ENTRY( 0x0040, NtGdiDdDDIQueryStatistics, 4 ) \
+    SYSCALL_ENTRY( 0x0041, NtGdiDdDDIQueryVideoMemoryInfo, 4 ) \
+    SYSCALL_ENTRY( 0x0042, NtGdiDdDDISetQueuedLimit, 4 ) \
+    SYSCALL_ENTRY( 0x0043, NtGdiDdDDISetVidPnSourceOwner, 4 ) \
+    SYSCALL_ENTRY( 0x0044, NtGdiDdDDIShareObjects, 20 ) \
+    SYSCALL_ENTRY( 0x0045, NtGdiDeleteClientObj, 4 ) \
+    SYSCALL_ENTRY( 0x0046, NtGdiDeleteObjectApp, 4 ) \
+    SYSCALL_ENTRY( 0x0047, NtGdiDescribePixelFormat, 16 ) \
+    SYSCALL_ENTRY( 0x0048, NtGdiDoPalette, 24 ) \
+    SYSCALL_ENTRY( 0x0049, NtGdiDrawStream, 12 ) \
+    SYSCALL_ENTRY( 0x004a, NtGdiEllipse, 20 ) \
+    SYSCALL_ENTRY( 0x004b, NtGdiEndDoc, 4 ) \
+    SYSCALL_ENTRY( 0x004c, NtGdiEndPage, 4 ) \
+    SYSCALL_ENTRY( 0x004d, NtGdiEndPath, 4 ) \
+    SYSCALL_ENTRY( 0x004e, NtGdiEnumFonts, 32 ) \
+    SYSCALL_ENTRY( 0x004f, NtGdiEqualRgn, 8 ) \
+    SYSCALL_ENTRY( 0x0050, NtGdiExcludeClipRect, 20 ) \
+    SYSCALL_ENTRY( 0x0051, NtGdiExtCreatePen, 44 ) \
+    SYSCALL_ENTRY( 0x0052, NtGdiExtCreateRegion, 12 ) \
+    SYSCALL_ENTRY( 0x0053, NtGdiExtEscape, 32 ) \
+    SYSCALL_ENTRY( 0x0054, NtGdiExtFloodFill, 20 ) \
+    SYSCALL_ENTRY( 0x0055, NtGdiExtGetObjectW, 12 ) \
+    SYSCALL_ENTRY( 0x0056, NtGdiExtSelectClipRgn, 12 ) \
+    SYSCALL_ENTRY( 0x0057, NtGdiExtTextOutW, 36 ) \
+    SYSCALL_ENTRY( 0x0058, NtGdiFillPath, 4 ) \
+    SYSCALL_ENTRY( 0x0059, NtGdiFillRgn, 12 ) \
+    SYSCALL_ENTRY( 0x005a, NtGdiFlattenPath, 4 ) \
+    SYSCALL_ENTRY( 0x005b, NtGdiFlush, 0 ) \
+    SYSCALL_ENTRY( 0x005c, NtGdiFontIsLinked, 4 ) \
+    SYSCALL_ENTRY( 0x005d, NtGdiFrameRgn, 20 ) \
+    SYSCALL_ENTRY( 0x005e, NtGdiGetAndSetDCDword, 16 ) \
+    SYSCALL_ENTRY( 0x005f, NtGdiGetAppClipBox, 8 ) \
+    SYSCALL_ENTRY( 0x0060, NtGdiGetBitmapBits, 12 ) \
+    SYSCALL_ENTRY( 0x0061, NtGdiGetBitmapDimension, 8 ) \
+    SYSCALL_ENTRY( 0x0062, NtGdiGetBoundsRect, 12 ) \
+    SYSCALL_ENTRY( 0x0063, NtGdiGetCharABCWidthsW, 24 ) \
+    SYSCALL_ENTRY( 0x0064, NtGdiGetCharWidthInfo, 8 ) \
+    SYSCALL_ENTRY( 0x0065, NtGdiGetCharWidthW, 24 ) \
+    SYSCALL_ENTRY( 0x0066, NtGdiGetColorAdjustment, 8 ) \
+    SYSCALL_ENTRY( 0x0067, NtGdiGetDCDword, 12 ) \
+    SYSCALL_ENTRY( 0x0068, NtGdiGetDCObject, 8 ) \
+    SYSCALL_ENTRY( 0x0069, NtGdiGetDCPoint, 12 ) \
+    SYSCALL_ENTRY( 0x006a, NtGdiGetDIBitsInternal, 36 ) \
+    SYSCALL_ENTRY( 0x006b, NtGdiGetDeviceCaps, 8 ) \
+    SYSCALL_ENTRY( 0x006c, NtGdiGetDeviceGammaRamp, 8 ) \
+    SYSCALL_ENTRY( 0x006d, NtGdiGetFontData, 20 ) \
+    SYSCALL_ENTRY( 0x006e, NtGdiGetFontFileData, 20 ) \
+    SYSCALL_ENTRY( 0x006f, NtGdiGetFontFileInfo, 20 ) \
+    SYSCALL_ENTRY( 0x0070, NtGdiGetFontUnicodeRanges, 8 ) \
+    SYSCALL_ENTRY( 0x0071, NtGdiGetGlyphIndicesW, 20 ) \
+    SYSCALL_ENTRY( 0x0072, NtGdiGetGlyphOutline, 32 ) \
+    SYSCALL_ENTRY( 0x0073, NtGdiGetKerningPairs, 12 ) \
+    SYSCALL_ENTRY( 0x0074, NtGdiGetNearestColor, 8 ) \
+    SYSCALL_ENTRY( 0x0075, NtGdiGetNearestPaletteIndex, 8 ) \
+    SYSCALL_ENTRY( 0x0076, NtGdiGetOutlineTextMetricsInternalW, 16 ) \
+    SYSCALL_ENTRY( 0x0077, NtGdiGetPath, 16 ) \
+    SYSCALL_ENTRY( 0x0078, NtGdiGetPixel, 12 ) \
+    SYSCALL_ENTRY( 0x0079, NtGdiGetRandomRgn, 12 ) \
+    SYSCALL_ENTRY( 0x007a, NtGdiGetRasterizerCaps, 8 ) \
+    SYSCALL_ENTRY( 0x007b, NtGdiGetRealizationInfo, 8 ) \
+    SYSCALL_ENTRY( 0x007c, NtGdiGetRegionData, 12 ) \
+    SYSCALL_ENTRY( 0x007d, NtGdiGetRgnBox, 8 ) \
+    SYSCALL_ENTRY( 0x007e, NtGdiGetSpoolMessage, 16 ) \
+    SYSCALL_ENTRY( 0x007f, NtGdiGetSystemPaletteUse, 4 ) \
+    SYSCALL_ENTRY( 0x0080, NtGdiGetTextCharsetInfo, 12 ) \
+    SYSCALL_ENTRY( 0x0081, NtGdiGetTextExtentExW, 32 ) \
+    SYSCALL_ENTRY( 0x0082, NtGdiGetTextFaceW, 16 ) \
+    SYSCALL_ENTRY( 0x0083, NtGdiGetTextMetricsW, 12 ) \
+    SYSCALL_ENTRY( 0x0084, NtGdiGetTransform, 12 ) \
+    SYSCALL_ENTRY( 0x0085, NtGdiGradientFill, 24 ) \
+    SYSCALL_ENTRY( 0x0086, NtGdiHfontCreate, 20 ) \
+    SYSCALL_ENTRY( 0x0087, NtGdiIcmBrushInfo, 32 ) \
+    SYSCALL_ENTRY( 0x0088, NtGdiInitSpool, 0 ) \
+    SYSCALL_ENTRY( 0x0089, NtGdiIntersectClipRect, 20 ) \
+    SYSCALL_ENTRY( 0x008a, NtGdiInvertRgn, 8 ) \
+    SYSCALL_ENTRY( 0x008b, NtGdiLineTo, 12 ) \
+    SYSCALL_ENTRY( 0x008c, NtGdiMaskBlt, 52 ) \
+    SYSCALL_ENTRY( 0x008d, NtGdiModifyWorldTransform, 12 ) \
+    SYSCALL_ENTRY( 0x008e, NtGdiMoveTo, 16 ) \
+    SYSCALL_ENTRY( 0x008f, NtGdiOffsetClipRgn, 12 ) \
+    SYSCALL_ENTRY( 0x0090, NtGdiOffsetRgn, 12 ) \
+    SYSCALL_ENTRY( 0x0091, NtGdiOpenDCW, 32 ) \
+    SYSCALL_ENTRY( 0x0092, NtGdiPatBlt, 24 ) \
+    SYSCALL_ENTRY( 0x0093, NtGdiPathToRegion, 4 ) \
+    SYSCALL_ENTRY( 0x0094, NtGdiPlgBlt, 44 ) \
+    SYSCALL_ENTRY( 0x0095, NtGdiPolyDraw, 16 ) \
+    SYSCALL_ENTRY( 0x0096, NtGdiPolyPolyDraw, 20 ) \
+    SYSCALL_ENTRY( 0x0097, NtGdiPtInRegion, 12 ) \
+    SYSCALL_ENTRY( 0x0098, NtGdiPtVisible, 12 ) \
+    SYSCALL_ENTRY( 0x0099, NtGdiRectInRegion, 8 ) \
+    SYSCALL_ENTRY( 0x009a, NtGdiRectVisible, 8 ) \
+    SYSCALL_ENTRY( 0x009b, NtGdiRectangle, 20 ) \
+    SYSCALL_ENTRY( 0x009c, NtGdiRemoveFontMemResourceEx, 4 ) \
+    SYSCALL_ENTRY( 0x009d, NtGdiRemoveFontResourceW, 24 ) \
+    SYSCALL_ENTRY( 0x009e, NtGdiResetDC, 20 ) \
+    SYSCALL_ENTRY( 0x009f, NtGdiResizePalette, 8 ) \
+    SYSCALL_ENTRY( 0x00a0, NtGdiRestoreDC, 8 ) \
+    SYSCALL_ENTRY( 0x00a1, NtGdiRoundRect, 28 ) \
+    SYSCALL_ENTRY( 0x00a2, NtGdiSaveDC, 4 ) \
+    SYSCALL_ENTRY( 0x00a3, NtGdiScaleViewportExtEx, 24 ) \
+    SYSCALL_ENTRY( 0x00a4, NtGdiScaleWindowExtEx, 24 ) \
+    SYSCALL_ENTRY( 0x00a5, NtGdiSelectBitmap, 8 ) \
+    SYSCALL_ENTRY( 0x00a6, NtGdiSelectBrush, 8 ) \
+    SYSCALL_ENTRY( 0x00a7, NtGdiSelectClipPath, 8 ) \
+    SYSCALL_ENTRY( 0x00a8, NtGdiSelectFont, 8 ) \
+    SYSCALL_ENTRY( 0x00a9, NtGdiSelectPen, 8 ) \
+    SYSCALL_ENTRY( 0x00aa, NtGdiSetBitmapBits, 12 ) \
+    SYSCALL_ENTRY( 0x00ab, NtGdiSetBitmapDimension, 16 ) \
+    SYSCALL_ENTRY( 0x00ac, NtGdiSetBoundsRect, 12 ) \
+    SYSCALL_ENTRY( 0x00ad, NtGdiSetBrushOrg, 16 ) \
+    SYSCALL_ENTRY( 0x00ae, NtGdiSetColorAdjustment, 8 ) \
+    SYSCALL_ENTRY( 0x00af, NtGdiSetDIBitsToDeviceInternal, 64 ) \
+    SYSCALL_ENTRY( 0x00b0, NtGdiSetDeviceGammaRamp, 8 ) \
+    SYSCALL_ENTRY( 0x00b1, NtGdiSetLayout, 12 ) \
+    SYSCALL_ENTRY( 0x00b2, NtGdiSetMagicColors, 12 ) \
+    SYSCALL_ENTRY( 0x00b3, NtGdiSetMetaRgn, 4 ) \
+    SYSCALL_ENTRY( 0x00b4, NtGdiSetPixel, 16 ) \
+    SYSCALL_ENTRY( 0x00b5, NtGdiSetPixelFormat, 8 ) \
+    SYSCALL_ENTRY( 0x00b6, NtGdiSetRectRgn, 20 ) \
+    SYSCALL_ENTRY( 0x00b7, NtGdiSetSystemPaletteUse, 8 ) \
+    SYSCALL_ENTRY( 0x00b8, NtGdiSetTextJustification, 12 ) \
+    SYSCALL_ENTRY( 0x00b9, NtGdiSetVirtualResolution, 20 ) \
+    SYSCALL_ENTRY( 0x00ba, NtGdiStartDoc, 16 ) \
+    SYSCALL_ENTRY( 0x00bb, NtGdiStartPage, 4 ) \
+    SYSCALL_ENTRY( 0x00bc, NtGdiStretchBlt, 48 ) \
+    SYSCALL_ENTRY( 0x00bd, NtGdiStretchDIBitsInternal, 64 ) \
+    SYSCALL_ENTRY( 0x00be, NtGdiStrokeAndFillPath, 4 ) \
+    SYSCALL_ENTRY( 0x00bf, NtGdiStrokePath, 4 ) \
+    SYSCALL_ENTRY( 0x00c0, NtGdiSwapBuffers, 4 ) \
+    SYSCALL_ENTRY( 0x00c1, NtGdiTransformPoints, 20 ) \
+    SYSCALL_ENTRY( 0x00c2, NtGdiTransparentBlt, 44 ) \
+    SYSCALL_ENTRY( 0x00c3, NtGdiUnrealizeObject, 4 ) \
+    SYSCALL_ENTRY( 0x00c4, NtGdiUpdateColors, 4 ) \
+    SYSCALL_ENTRY( 0x00c5, NtGdiWidenPath, 4 ) \
+    SYSCALL_ENTRY( 0x00c6, NtUserActivateKeyboardLayout, 8 ) \
+    SYSCALL_ENTRY( 0x00c7, NtUserAddClipboardFormatListener, 4 ) \
+    SYSCALL_ENTRY( 0x00c8, NtUserAssociateInputContext, 12 ) \
+    SYSCALL_ENTRY( 0x00c9, NtUserAttachThreadInput, 12 ) \
+    SYSCALL_ENTRY( 0x00ca, NtUserBeginPaint, 8 ) \
+    SYSCALL_ENTRY( 0x00cb, NtUserBuildHimcList, 16 ) \
+    SYSCALL_ENTRY( 0x00cc, NtUserBuildHwndList, 32 ) \
+    SYSCALL_ENTRY( 0x00cd, NtUserBuildNameList, 16 ) \
+    SYSCALL_ENTRY( 0x00ce, NtUserBuildPropList, 16 ) \
+    SYSCALL_ENTRY( 0x00cf, NtUserCallHwnd, 8 ) \
+    SYSCALL_ENTRY( 0x00d0, NtUserCallHwndParam, 12 ) \
+    SYSCALL_ENTRY( 0x00d1, NtUserCallMsgFilter, 8 ) \
+    SYSCALL_ENTRY( 0x00d2, NtUserCallNextHookEx, 16 ) \
+    SYSCALL_ENTRY( 0x00d3, NtUserCallNoParam, 4 ) \
+    SYSCALL_ENTRY( 0x00d4, NtUserCallOneParam, 8 ) \
+    SYSCALL_ENTRY( 0x00d5, NtUserCallTwoParam, 12 ) \
+    SYSCALL_ENTRY( 0x00d6, NtUserChangeClipboardChain, 8 ) \
+    SYSCALL_ENTRY( 0x00d7, NtUserChangeDisplaySettings, 20 ) \
+    SYSCALL_ENTRY( 0x00d8, NtUserCheckMenuItem, 12 ) \
+    SYSCALL_ENTRY( 0x00d9, NtUserChildWindowFromPointEx, 16 ) \
+    SYSCALL_ENTRY( 0x00da, NtUserClipCursor, 4 ) \
+    SYSCALL_ENTRY( 0x00db, NtUserCloseClipboard, 0 ) \
+    SYSCALL_ENTRY( 0x00dc, NtUserCloseDesktop, 4 ) \
+    SYSCALL_ENTRY( 0x00dd, NtUserCloseWindowStation, 4 ) \
+    SYSCALL_ENTRY( 0x00de, NtUserCopyAcceleratorTable, 12 ) \
+    SYSCALL_ENTRY( 0x00df, NtUserCountClipboardFormats, 0 ) \
+    SYSCALL_ENTRY( 0x00e0, NtUserCreateAcceleratorTable, 8 ) \
+    SYSCALL_ENTRY( 0x00e1, NtUserCreateCaret, 16 ) \
+    SYSCALL_ENTRY( 0x00e2, NtUserCreateDesktopEx, 24 ) \
+    SYSCALL_ENTRY( 0x00e3, NtUserCreateInputContext, 4 ) \
+    SYSCALL_ENTRY( 0x00e4, NtUserCreateMenu, 0 ) \
+    SYSCALL_ENTRY( 0x00e5, NtUserCreatePopupMenu, 0 ) \
+    SYSCALL_ENTRY( 0x00e6, NtUserCreateWindowEx, 68 ) \
+    SYSCALL_ENTRY( 0x00e7, NtUserCreateWindowStation, 28 ) \
+    SYSCALL_ENTRY( 0x00e8, NtUserDeferWindowPosAndBand, 40 ) \
+    SYSCALL_ENTRY( 0x00e9, NtUserDeleteMenu, 12 ) \
+    SYSCALL_ENTRY( 0x00ea, NtUserDestroyAcceleratorTable, 4 ) \
+    SYSCALL_ENTRY( 0x00eb, NtUserDestroyCaret, 0 ) \
+    SYSCALL_ENTRY( 0x00ec, NtUserDestroyCursor, 8 ) \
+    SYSCALL_ENTRY( 0x00ed, NtUserDestroyInputContext, 4 ) \
+    SYSCALL_ENTRY( 0x00ee, NtUserDestroyMenu, 4 ) \
+    SYSCALL_ENTRY( 0x00ef, NtUserDestroyWindow, 4 ) \
+    SYSCALL_ENTRY( 0x00f0, NtUserDisableThreadIme, 4 ) \
+    SYSCALL_ENTRY( 0x00f1, NtUserDispatchMessage, 4 ) \
+    SYSCALL_ENTRY( 0x00f2, NtUserDisplayConfigGetDeviceInfo, 4 ) \
+    SYSCALL_ENTRY( 0x00f3, NtUserDragDetect, 12 ) \
+    SYSCALL_ENTRY( 0x00f4, NtUserDragObject, 20 ) \
+    SYSCALL_ENTRY( 0x00f5, NtUserDrawCaptionTemp, 28 ) \
+    SYSCALL_ENTRY( 0x00f6, NtUserDrawIconEx, 36 ) \
+    SYSCALL_ENTRY( 0x00f7, NtUserDrawMenuBarTemp, 20 ) \
+    SYSCALL_ENTRY( 0x00f8, NtUserEmptyClipboard, 0 ) \
+    SYSCALL_ENTRY( 0x00f9, NtUserEnableMenuItem, 12 ) \
+    SYSCALL_ENTRY( 0x00fa, NtUserEnableMouseInPointer, 4 ) \
+    SYSCALL_ENTRY( 0x00fb, NtUserEnableMouseInPointerForThread, 0 ) \
+    SYSCALL_ENTRY( 0x00fc, NtUserEnableScrollBar, 12 ) \
+    SYSCALL_ENTRY( 0x00fd, NtUserEndDeferWindowPosEx, 8 ) \
+    SYSCALL_ENTRY( 0x00fe, NtUserEndMenu, 0 ) \
+    SYSCALL_ENTRY( 0x00ff, NtUserEndPaint, 8 ) \
+    SYSCALL_ENTRY( 0x0100, NtUserEnumClipboardFormats, 4 ) \
+    SYSCALL_ENTRY( 0x0101, NtUserEnumDisplayDevices, 16 ) \
+    SYSCALL_ENTRY( 0x0102, NtUserEnumDisplayMonitors, 16 ) \
+    SYSCALL_ENTRY( 0x0103, NtUserEnumDisplaySettings, 16 ) \
+    SYSCALL_ENTRY( 0x0104, NtUserExcludeUpdateRgn, 8 ) \
+    SYSCALL_ENTRY( 0x0105, NtUserFindExistingCursorIcon, 12 ) \
+    SYSCALL_ENTRY( 0x0106, NtUserFindWindowEx, 20 ) \
+    SYSCALL_ENTRY( 0x0107, NtUserFlashWindowEx, 4 ) \
+    SYSCALL_ENTRY( 0x0108, NtUserGetAncestor, 8 ) \
+    SYSCALL_ENTRY( 0x0109, NtUserGetAsyncKeyState, 4 ) \
+    SYSCALL_ENTRY( 0x010a, NtUserGetAtomName, 8 ) \
+    SYSCALL_ENTRY( 0x010b, NtUserGetCaretBlinkTime, 0 ) \
+    SYSCALL_ENTRY( 0x010c, NtUserGetCaretPos, 4 ) \
+    SYSCALL_ENTRY( 0x010d, NtUserGetClassInfoEx, 20 ) \
+    SYSCALL_ENTRY( 0x010e, NtUserGetClassName, 12 ) \
+    SYSCALL_ENTRY( 0x010f, NtUserGetClipboardData, 8 ) \
+    SYSCALL_ENTRY( 0x0110, NtUserGetClipboardFormatName, 12 ) \
+    SYSCALL_ENTRY( 0x0111, NtUserGetClipboardOwner, 0 ) \
+    SYSCALL_ENTRY( 0x0112, NtUserGetClipboardSequenceNumber, 0 ) \
+    SYSCALL_ENTRY( 0x0113, NtUserGetClipboardViewer, 0 ) \
+    SYSCALL_ENTRY( 0x0114, NtUserGetCurrentInputMessageSource, 4 ) \
+    SYSCALL_ENTRY( 0x0115, NtUserGetCursor, 0 ) \
+    SYSCALL_ENTRY( 0x0116, NtUserGetCursorFrameInfo, 16 ) \
+    SYSCALL_ENTRY( 0x0117, NtUserGetCursorInfo, 4 ) \
+    SYSCALL_ENTRY( 0x0118, NtUserGetDC, 4 ) \
+    SYSCALL_ENTRY( 0x0119, NtUserGetDCEx, 12 ) \
+    SYSCALL_ENTRY( 0x011a, NtUserGetDisplayConfigBufferSizes, 12 ) \
+    SYSCALL_ENTRY( 0x011b, NtUserGetDoubleClickTime, 0 ) \
+    SYSCALL_ENTRY( 0x011c, NtUserGetDpiForMonitor, 16 ) \
+    SYSCALL_ENTRY( 0x011d, NtUserGetForegroundWindow, 0 ) \
+    SYSCALL_ENTRY( 0x011e, NtUserGetGUIThreadInfo, 8 ) \
+    SYSCALL_ENTRY( 0x011f, NtUserGetIconInfo, 24 ) \
+    SYSCALL_ENTRY( 0x0120, NtUserGetIconSize, 16 ) \
+    SYSCALL_ENTRY( 0x0121, NtUserGetInternalWindowPos, 12 ) \
+    SYSCALL_ENTRY( 0x0122, NtUserGetKeyNameText, 12 ) \
+    SYSCALL_ENTRY( 0x0123, NtUserGetKeyState, 4 ) \
+    SYSCALL_ENTRY( 0x0124, NtUserGetKeyboardLayout, 4 ) \
+    SYSCALL_ENTRY( 0x0125, NtUserGetKeyboardLayoutList, 8 ) \
+    SYSCALL_ENTRY( 0x0126, NtUserGetKeyboardLayoutName, 4 ) \
+    SYSCALL_ENTRY( 0x0127, NtUserGetKeyboardState, 4 ) \
+    SYSCALL_ENTRY( 0x0128, NtUserGetLayeredWindowAttributes, 16 ) \
+    SYSCALL_ENTRY( 0x0129, NtUserGetMenuBarInfo, 16 ) \
+    SYSCALL_ENTRY( 0x012a, NtUserGetMenuItemRect, 16 ) \
+    SYSCALL_ENTRY( 0x012b, NtUserGetMessage, 16 ) \
+    SYSCALL_ENTRY( 0x012c, NtUserGetMouseMovePointsEx, 20 ) \
+    SYSCALL_ENTRY( 0x012d, NtUserGetObjectInformation, 20 ) \
+    SYSCALL_ENTRY( 0x012e, NtUserGetOpenClipboardWindow, 0 ) \
+    SYSCALL_ENTRY( 0x012f, NtUserGetPointerInfoList, 32 ) \
+    SYSCALL_ENTRY( 0x0130, NtUserGetPriorityClipboardFormat, 8 ) \
+    SYSCALL_ENTRY( 0x0131, NtUserGetProcessDpiAwarenessContext, 4 ) \
+    SYSCALL_ENTRY( 0x0132, NtUserGetProcessWindowStation, 0 ) \
+    SYSCALL_ENTRY( 0x0133, NtUserGetProp, 8 ) \
+    SYSCALL_ENTRY( 0x0134, NtUserGetQueueStatus, 4 ) \
+    SYSCALL_ENTRY( 0x0135, NtUserGetRawInputBuffer, 12 ) \
+    SYSCALL_ENTRY( 0x0136, NtUserGetRawInputData, 20 ) \
+    SYSCALL_ENTRY( 0x0137, NtUserGetRawInputDeviceInfo, 16 ) \
+    SYSCALL_ENTRY( 0x0138, NtUserGetRawInputDeviceList, 12 ) \
+    SYSCALL_ENTRY( 0x0139, NtUserGetRegisteredRawInputDevices, 12 ) \
+    SYSCALL_ENTRY( 0x013a, NtUserGetScrollBarInfo, 12 ) \
+    SYSCALL_ENTRY( 0x013b, NtUserGetSystemDpiForProcess, 4 ) \
+    SYSCALL_ENTRY( 0x013c, NtUserGetSystemMenu, 8 ) \
+    SYSCALL_ENTRY( 0x013d, NtUserGetThreadDesktop, 4 ) \
+    SYSCALL_ENTRY( 0x013e, NtUserGetThreadState, 4 ) \
+    SYSCALL_ENTRY( 0x013f, NtUserGetTitleBarInfo, 8 ) \
+    SYSCALL_ENTRY( 0x0140, NtUserGetTouchInputInfo, 16 ) \
+    SYSCALL_ENTRY( 0x0141, NtUserGetUpdateRect, 12 ) \
+    SYSCALL_ENTRY( 0x0142, NtUserGetUpdateRgn, 12 ) \
+    SYSCALL_ENTRY( 0x0143, NtUserGetUpdatedClipboardFormats, 12 ) \
+    SYSCALL_ENTRY( 0x0144, NtUserGetWindowDC, 4 ) \
+    SYSCALL_ENTRY( 0x0145, NtUserGetWindowPlacement, 8 ) \
+    SYSCALL_ENTRY( 0x0146, NtUserGetWindowRgnEx, 12 ) \
+    SYSCALL_ENTRY( 0x0147, NtUserHideCaret, 4 ) \
+    SYSCALL_ENTRY( 0x0148, NtUserHiliteMenuItem, 16 ) \
+    SYSCALL_ENTRY( 0x0149, NtUserInitializeClientPfnArrays, 16 ) \
+    SYSCALL_ENTRY( 0x014a, NtUserInternalGetWindowIcon, 8 ) \
+    SYSCALL_ENTRY( 0x014b, NtUserInternalGetWindowText, 12 ) \
+    SYSCALL_ENTRY( 0x014c, NtUserInvalidateRect, 12 ) \
+    SYSCALL_ENTRY( 0x014d, NtUserInvalidateRgn, 12 ) \
+    SYSCALL_ENTRY( 0x014e, NtUserIsClipboardFormatAvailable, 4 ) \
+    SYSCALL_ENTRY( 0x014f, NtUserIsMouseInPointerEnabled, 0 ) \
+    SYSCALL_ENTRY( 0x0150, NtUserIsTouchWindow, 8 ) \
+    SYSCALL_ENTRY( 0x0151, NtUserKillTimer, 8 ) \
+    SYSCALL_ENTRY( 0x0152, NtUserLockWindowUpdate, 4 ) \
+    SYSCALL_ENTRY( 0x0153, NtUserLogicalToPerMonitorDPIPhysicalPoint, 8 ) \
+    SYSCALL_ENTRY( 0x0154, NtUserMapVirtualKeyEx, 12 ) \
+    SYSCALL_ENTRY( 0x0155, NtUserMenuItemFromPoint, 16 ) \
+    SYSCALL_ENTRY( 0x0156, NtUserMessageBeep, 4 ) \
+    SYSCALL_ENTRY( 0x0157, NtUserMessageCall, 28 ) \
+    SYSCALL_ENTRY( 0x0158, NtUserMoveWindow, 24 ) \
+    SYSCALL_ENTRY( 0x0159, NtUserMsgWaitForMultipleObjectsEx, 20 ) \
+    SYSCALL_ENTRY( 0x015a, NtUserNotifyIMEStatus, 8 ) \
+    SYSCALL_ENTRY( 0x015b, NtUserNotifyWinEvent, 16 ) \
+    SYSCALL_ENTRY( 0x015c, NtUserOpenClipboard, 8 ) \
+    SYSCALL_ENTRY( 0x015d, NtUserOpenDesktop, 12 ) \
+    SYSCALL_ENTRY( 0x015e, NtUserOpenInputDesktop, 12 ) \
+    SYSCALL_ENTRY( 0x015f, NtUserOpenWindowStation, 8 ) \
+    SYSCALL_ENTRY( 0x0160, NtUserPeekMessage, 20 ) \
+    SYSCALL_ENTRY( 0x0161, NtUserPerMonitorDPIPhysicalToLogicalPoint, 8 ) \
+    SYSCALL_ENTRY( 0x0162, NtUserPostMessage, 16 ) \
+    SYSCALL_ENTRY( 0x0163, NtUserPostQuitMessage, 4 ) \
+    SYSCALL_ENTRY( 0x0164, NtUserPostThreadMessage, 16 ) \
+    SYSCALL_ENTRY( 0x0165, NtUserPrintWindow, 12 ) \
+    SYSCALL_ENTRY( 0x0166, NtUserQueryDisplayConfig, 24 ) \
+    SYSCALL_ENTRY( 0x0167, NtUserQueryInputContext, 8 ) \
+    SYSCALL_ENTRY( 0x0168, NtUserQueryWindow, 8 ) \
+    SYSCALL_ENTRY( 0x0169, NtUserRealChildWindowFromPoint, 12 ) \
+    SYSCALL_ENTRY( 0x016a, NtUserRedrawWindow, 16 ) \
+    SYSCALL_ENTRY( 0x016b, NtUserRegisterClassExWOW, 28 ) \
+    SYSCALL_ENTRY( 0x016c, NtUserRegisterHotKey, 16 ) \
+    SYSCALL_ENTRY( 0x016d, NtUserRegisterRawInputDevices, 12 ) \
+    SYSCALL_ENTRY( 0x016e, NtUserRegisterTouchPadCapable, 4 ) \
+    SYSCALL_ENTRY( 0x016f, NtUserReleaseCapture, 0 ) \
+    SYSCALL_ENTRY( 0x0170, NtUserReleaseDC, 8 ) \
+    SYSCALL_ENTRY( 0x0171, NtUserRemoveClipboardFormatListener, 4 ) \
+    SYSCALL_ENTRY( 0x0172, NtUserRemoveMenu, 12 ) \
+    SYSCALL_ENTRY( 0x0173, NtUserRemoveProp, 8 ) \
+    SYSCALL_ENTRY( 0x0174, NtUserScrollDC, 28 ) \
+    SYSCALL_ENTRY( 0x0175, NtUserScrollWindowEx, 32 ) \
+    SYSCALL_ENTRY( 0x0176, NtUserSelectPalette, 12 ) \
+    SYSCALL_ENTRY( 0x0177, NtUserSendInput, 12 ) \
+    SYSCALL_ENTRY( 0x0178, NtUserSetActiveWindow, 4 ) \
+    SYSCALL_ENTRY( 0x0179, NtUserSetAdditionalForegroundBoostProcesses, 12 ) \
+    SYSCALL_ENTRY( 0x017a, NtUserSetCapture, 4 ) \
+    SYSCALL_ENTRY( 0x017b, NtUserSetClassLong, 16 ) \
+    SYSCALL_ENTRY( 0x017c, NtUserSetClassLongPtr, 16 ) \
+    SYSCALL_ENTRY( 0x017d, NtUserSetClassWord, 12 ) \
+    SYSCALL_ENTRY( 0x017e, NtUserSetClipboardData, 12 ) \
+    SYSCALL_ENTRY( 0x017f, NtUserSetClipboardViewer, 4 ) \
+    SYSCALL_ENTRY( 0x0180, NtUserSetCursor, 4 ) \
+    SYSCALL_ENTRY( 0x0181, NtUserSetCursorIconData, 16 ) \
+    SYSCALL_ENTRY( 0x0182, NtUserSetCursorPos, 8 ) \
+    SYSCALL_ENTRY( 0x0183, NtUserSetFocus, 4 ) \
+    SYSCALL_ENTRY( 0x0184, NtUserSetInternalWindowPos, 16 ) \
+    SYSCALL_ENTRY( 0x0185, NtUserSetKeyboardState, 4 ) \
+    SYSCALL_ENTRY( 0x0186, NtUserSetLayeredWindowAttributes, 16 ) \
+    SYSCALL_ENTRY( 0x0187, NtUserSetMenu, 8 ) \
+    SYSCALL_ENTRY( 0x0188, NtUserSetMenuContextHelpId, 8 ) \
+    SYSCALL_ENTRY( 0x0189, NtUserSetMenuDefaultItem, 12 ) \
+    SYSCALL_ENTRY( 0x018a, NtUserSetObjectInformation, 16 ) \
+    SYSCALL_ENTRY( 0x018b, NtUserSetParent, 8 ) \
+    SYSCALL_ENTRY( 0x018c, NtUserSetProcessDpiAwarenessContext, 8 ) \
+    SYSCALL_ENTRY( 0x018d, NtUserSetProcessWindowStation, 4 ) \
+    SYSCALL_ENTRY( 0x018e, NtUserSetProp, 12 ) \
+    SYSCALL_ENTRY( 0x018f, NtUserSetScrollInfo, 16 ) \
+    SYSCALL_ENTRY( 0x0190, NtUserSetShellWindowEx, 8 ) \
+    SYSCALL_ENTRY( 0x0191, NtUserSetSysColors, 12 ) \
+    SYSCALL_ENTRY( 0x0192, NtUserSetSystemMenu, 8 ) \
+    SYSCALL_ENTRY( 0x0193, NtUserSetSystemTimer, 12 ) \
+    SYSCALL_ENTRY( 0x0194, NtUserSetThreadDesktop, 4 ) \
+    SYSCALL_ENTRY( 0x0195, NtUserSetTimer, 20 ) \
+    SYSCALL_ENTRY( 0x0196, NtUserSetWinEventHook, 32 ) \
+    SYSCALL_ENTRY( 0x0197, NtUserSetWindowLong, 16 ) \
+    SYSCALL_ENTRY( 0x0198, NtUserSetWindowLongPtr, 16 ) \
+    SYSCALL_ENTRY( 0x0199, NtUserSetWindowPlacement, 8 ) \
+    SYSCALL_ENTRY( 0x019a, NtUserSetWindowPos, 28 ) \
+    SYSCALL_ENTRY( 0x019b, NtUserSetWindowRgn, 12 ) \
+    SYSCALL_ENTRY( 0x019c, NtUserSetWindowWord, 12 ) \
+    SYSCALL_ENTRY( 0x019d, NtUserSetWindowsHookEx, 24 ) \
+    SYSCALL_ENTRY( 0x019e, NtUserShowCaret, 4 ) \
+    SYSCALL_ENTRY( 0x019f, NtUserShowCursor, 4 ) \
+    SYSCALL_ENTRY( 0x01a0, NtUserShowScrollBar, 12 ) \
+    SYSCALL_ENTRY( 0x01a1, NtUserShowWindow, 8 ) \
+    SYSCALL_ENTRY( 0x01a2, NtUserShowWindowAsync, 8 ) \
+    SYSCALL_ENTRY( 0x01a3, NtUserSwitchDesktop, 4 ) \
+    SYSCALL_ENTRY( 0x01a4, NtUserSystemParametersInfo, 16 ) \
+    SYSCALL_ENTRY( 0x01a5, NtUserSystemParametersInfoForDpi, 20 ) \
+    SYSCALL_ENTRY( 0x01a6, NtUserThunkedMenuInfo, 8 ) \
+    SYSCALL_ENTRY( 0x01a7, NtUserThunkedMenuItemInfo, 24 ) \
+    SYSCALL_ENTRY( 0x01a8, NtUserToUnicodeEx, 28 ) \
+    SYSCALL_ENTRY( 0x01a9, NtUserTrackMouseEvent, 4 ) \
+    SYSCALL_ENTRY( 0x01aa, NtUserTrackPopupMenuEx, 24 ) \
+    SYSCALL_ENTRY( 0x01ab, NtUserTranslateAccelerator, 12 ) \
+    SYSCALL_ENTRY( 0x01ac, NtUserTranslateMessage, 8 ) \
+    SYSCALL_ENTRY( 0x01ad, NtUserUnhookWinEvent, 4 ) \
+    SYSCALL_ENTRY( 0x01ae, NtUserUnhookWindowsHookEx, 4 ) \
+    SYSCALL_ENTRY( 0x01af, NtUserUnregisterClass, 12 ) \
+    SYSCALL_ENTRY( 0x01b0, NtUserUnregisterHotKey, 8 ) \
+    SYSCALL_ENTRY( 0x01b1, NtUserUpdateInputContext, 12 ) \
+    SYSCALL_ENTRY( 0x01b2, NtUserUpdateLayeredWindow, 40 ) \
+    SYSCALL_ENTRY( 0x01b3, NtUserValidateRect, 8 ) \
+    SYSCALL_ENTRY( 0x01b4, NtUserVkKeyScanEx, 8 ) \
+    SYSCALL_ENTRY( 0x01b5, NtUserWaitForInputIdle, 12 ) \
+    SYSCALL_ENTRY( 0x01b6, NtUserWaitMessage, 0 ) \
+    SYSCALL_ENTRY( 0x01b7, NtUserWindowFromDC, 4 ) \
+    SYSCALL_ENTRY( 0x01b8, NtUserWindowFromPoint, 8 ) \
+    SYSCALL_ENTRY( 0x01b9, __wine_get_file_outline_text_metric, 16 ) \
+    SYSCALL_ENTRY( 0x01ba, __wine_get_icm_profile, 16 )
+
+#define ALL_SYSCALLS64 \
+    SYSCALL_ENTRY( 0x0000, NtGdiAbortDoc, 8 ) \
+    SYSCALL_ENTRY( 0x0001, NtGdiAbortPath, 8 ) \
+    SYSCALL_ENTRY( 0x0002, NtGdiAddFontMemResourceEx, 40 ) \
+    SYSCALL_ENTRY( 0x0003, NtGdiAddFontResourceW, 48 ) \
+    SYSCALL_ENTRY( 0x0004, NtGdiAlphaBlend, 96 ) \
+    SYSCALL_ENTRY( 0x0005, NtGdiAngleArc, 48 ) \
+    SYSCALL_ENTRY( 0x0006, NtGdiArcInternal, 80 ) \
+    SYSCALL_ENTRY( 0x0007, NtGdiBeginPath, 8 ) \
+    SYSCALL_ENTRY( 0x0008, NtGdiBitBlt, 88 ) \
+    SYSCALL_ENTRY( 0x0009, NtGdiCloseFigure, 8 ) \
+    SYSCALL_ENTRY( 0x000a, NtGdiCombineRgn, 32 ) \
+    SYSCALL_ENTRY( 0x000b, NtGdiComputeXformCoefficients, 8 ) \
+    SYSCALL_ENTRY( 0x000c, NtGdiCreateBitmap, 40 ) \
+    SYSCALL_ENTRY( 0x000d, NtGdiCreateClientObj, 8 ) \
+    SYSCALL_ENTRY( 0x000e, NtGdiCreateCompatibleBitmap, 24 ) \
+    SYSCALL_ENTRY( 0x000f, NtGdiCreateCompatibleDC, 8 ) \
+    SYSCALL_ENTRY( 0x0010, NtGdiCreateDIBBrush, 48 ) \
+    SYSCALL_ENTRY( 0x0011, NtGdiCreateDIBSection, 72 ) \
+    SYSCALL_ENTRY( 0x0012, NtGdiCreateDIBitmapInternal, 88 ) \
+    SYSCALL_ENTRY( 0x0013, NtGdiCreateEllipticRgn, 32 ) \
+    SYSCALL_ENTRY( 0x0014, NtGdiCreateHalftonePalette, 8 ) \
+    SYSCALL_ENTRY( 0x0015, NtGdiCreateHatchBrushInternal, 24 ) \
+    SYSCALL_ENTRY( 0x0016, NtGdiCreateMetafileDC, 8 ) \
+    SYSCALL_ENTRY( 0x0017, NtGdiCreatePaletteInternal, 16 ) \
+    SYSCALL_ENTRY( 0x0018, NtGdiCreatePatternBrushInternal, 24 ) \
+    SYSCALL_ENTRY( 0x0019, NtGdiCreatePen, 32 ) \
+    SYSCALL_ENTRY( 0x001a, NtGdiCreateRectRgn, 32 ) \
+    SYSCALL_ENTRY( 0x001b, NtGdiCreateRoundRectRgn, 48 ) \
+    SYSCALL_ENTRY( 0x001c, NtGdiCreateSolidBrush, 16 ) \
+    SYSCALL_ENTRY( 0x001d, NtGdiDdDDICheckVidPnExclusiveOwnership, 8 ) \
+    SYSCALL_ENTRY( 0x001e, NtGdiDdDDICloseAdapter, 8 ) \
+    SYSCALL_ENTRY( 0x001f, NtGdiDdDDICreateAllocation, 8 ) \
+    SYSCALL_ENTRY( 0x0020, NtGdiDdDDICreateAllocation2, 8 ) \
+    SYSCALL_ENTRY( 0x0021, NtGdiDdDDICreateDCFromMemory, 8 ) \
+    SYSCALL_ENTRY( 0x0022, NtGdiDdDDICreateDevice, 8 ) \
+    SYSCALL_ENTRY( 0x0023, NtGdiDdDDICreateKeyedMutex, 8 ) \
+    SYSCALL_ENTRY( 0x0024, NtGdiDdDDICreateKeyedMutex2, 8 ) \
+    SYSCALL_ENTRY( 0x0025, NtGdiDdDDICreateSynchronizationObject, 8 ) \
+    SYSCALL_ENTRY( 0x0026, NtGdiDdDDICreateSynchronizationObject2, 8 ) \
+    SYSCALL_ENTRY( 0x0027, NtGdiDdDDIDestroyAllocation, 8 ) \
+    SYSCALL_ENTRY( 0x0028, NtGdiDdDDIDestroyAllocation2, 8 ) \
+    SYSCALL_ENTRY( 0x0029, NtGdiDdDDIDestroyDCFromMemory, 8 ) \
+    SYSCALL_ENTRY( 0x002a, NtGdiDdDDIDestroyDevice, 8 ) \
+    SYSCALL_ENTRY( 0x002b, NtGdiDdDDIDestroyKeyedMutex, 8 ) \
+    SYSCALL_ENTRY( 0x002c, NtGdiDdDDIDestroySynchronizationObject, 8 ) \
+    SYSCALL_ENTRY( 0x002d, NtGdiDdDDIEnumAdapters, 8 ) \
+    SYSCALL_ENTRY( 0x002e, NtGdiDdDDIEnumAdapters2, 8 ) \
+    SYSCALL_ENTRY( 0x002f, NtGdiDdDDIEscape, 8 ) \
+    SYSCALL_ENTRY( 0x0030, NtGdiDdDDIOpenAdapterFromDeviceName, 8 ) \
+    SYSCALL_ENTRY( 0x0031, NtGdiDdDDIOpenAdapterFromHdc, 8 ) \
+    SYSCALL_ENTRY( 0x0032, NtGdiDdDDIOpenAdapterFromLuid, 8 ) \
+    SYSCALL_ENTRY( 0x0033, NtGdiDdDDIOpenKeyedMutex, 8 ) \
+    SYSCALL_ENTRY( 0x0034, NtGdiDdDDIOpenKeyedMutex2, 8 ) \
+    SYSCALL_ENTRY( 0x0035, NtGdiDdDDIOpenKeyedMutexFromNtHandle, 8 ) \
+    SYSCALL_ENTRY( 0x0036, NtGdiDdDDIOpenResource, 8 ) \
+    SYSCALL_ENTRY( 0x0037, NtGdiDdDDIOpenResource2, 8 ) \
+    SYSCALL_ENTRY( 0x0038, NtGdiDdDDIOpenResourceFromNtHandle, 8 ) \
+    SYSCALL_ENTRY( 0x0039, NtGdiDdDDIOpenSyncObjectFromNtHandle, 8 ) \
+    SYSCALL_ENTRY( 0x003a, NtGdiDdDDIOpenSyncObjectFromNtHandle2, 8 ) \
+    SYSCALL_ENTRY( 0x003b, NtGdiDdDDIOpenSyncObjectNtHandleFromName, 8 ) \
+    SYSCALL_ENTRY( 0x003c, NtGdiDdDDIOpenSynchronizationObject, 8 ) \
+    SYSCALL_ENTRY( 0x003d, NtGdiDdDDIQueryAdapterInfo, 8 ) \
+    SYSCALL_ENTRY( 0x003e, NtGdiDdDDIQueryResourceInfo, 8 ) \
+    SYSCALL_ENTRY( 0x003f, NtGdiDdDDIQueryResourceInfoFromNtHandle, 8 ) \
+    SYSCALL_ENTRY( 0x0040, NtGdiDdDDIQueryStatistics, 8 ) \
+    SYSCALL_ENTRY( 0x0041, NtGdiDdDDIQueryVideoMemoryInfo, 8 ) \
+    SYSCALL_ENTRY( 0x0042, NtGdiDdDDISetQueuedLimit, 8 ) \
+    SYSCALL_ENTRY( 0x0043, NtGdiDdDDISetVidPnSourceOwner, 8 ) \
+    SYSCALL_ENTRY( 0x0044, NtGdiDdDDIShareObjects, 40 ) \
+    SYSCALL_ENTRY( 0x0045, NtGdiDeleteClientObj, 8 ) \
+    SYSCALL_ENTRY( 0x0046, NtGdiDeleteObjectApp, 8 ) \
+    SYSCALL_ENTRY( 0x0047, NtGdiDescribePixelFormat, 32 ) \
+    SYSCALL_ENTRY( 0x0048, NtGdiDoPalette, 48 ) \
+    SYSCALL_ENTRY( 0x0049, NtGdiDrawStream, 24 ) \
+    SYSCALL_ENTRY( 0x004a, NtGdiEllipse, 40 ) \
+    SYSCALL_ENTRY( 0x004b, NtGdiEndDoc, 8 ) \
+    SYSCALL_ENTRY( 0x004c, NtGdiEndPage, 8 ) \
+    SYSCALL_ENTRY( 0x004d, NtGdiEndPath, 8 ) \
+    SYSCALL_ENTRY( 0x004e, NtGdiEnumFonts, 64 ) \
+    SYSCALL_ENTRY( 0x004f, NtGdiEqualRgn, 16 ) \
+    SYSCALL_ENTRY( 0x0050, NtGdiExcludeClipRect, 40 ) \
+    SYSCALL_ENTRY( 0x0051, NtGdiExtCreatePen, 88 ) \
+    SYSCALL_ENTRY( 0x0052, NtGdiExtCreateRegion, 24 ) \
+    SYSCALL_ENTRY( 0x0053, NtGdiExtEscape, 64 ) \
+    SYSCALL_ENTRY( 0x0054, NtGdiExtFloodFill, 40 ) \
+    SYSCALL_ENTRY( 0x0055, NtGdiExtGetObjectW, 24 ) \
+    SYSCALL_ENTRY( 0x0056, NtGdiExtSelectClipRgn, 24 ) \
+    SYSCALL_ENTRY( 0x0057, NtGdiExtTextOutW, 72 ) \
+    SYSCALL_ENTRY( 0x0058, NtGdiFillPath, 8 ) \
+    SYSCALL_ENTRY( 0x0059, NtGdiFillRgn, 24 ) \
+    SYSCALL_ENTRY( 0x005a, NtGdiFlattenPath, 8 ) \
+    SYSCALL_ENTRY( 0x005b, NtGdiFlush, 0 ) \
+    SYSCALL_ENTRY( 0x005c, NtGdiFontIsLinked, 8 ) \
+    SYSCALL_ENTRY( 0x005d, NtGdiFrameRgn, 40 ) \
+    SYSCALL_ENTRY( 0x005e, NtGdiGetAndSetDCDword, 32 ) \
+    SYSCALL_ENTRY( 0x005f, NtGdiGetAppClipBox, 16 ) \
+    SYSCALL_ENTRY( 0x0060, NtGdiGetBitmapBits, 24 ) \
+    SYSCALL_ENTRY( 0x0061, NtGdiGetBitmapDimension, 16 ) \
+    SYSCALL_ENTRY( 0x0062, NtGdiGetBoundsRect, 24 ) \
+    SYSCALL_ENTRY( 0x0063, NtGdiGetCharABCWidthsW, 48 ) \
+    SYSCALL_ENTRY( 0x0064, NtGdiGetCharWidthInfo, 16 ) \
+    SYSCALL_ENTRY( 0x0065, NtGdiGetCharWidthW, 48 ) \
+    SYSCALL_ENTRY( 0x0066, NtGdiGetColorAdjustment, 16 ) \
+    SYSCALL_ENTRY( 0x0067, NtGdiGetDCDword, 24 ) \
+    SYSCALL_ENTRY( 0x0068, NtGdiGetDCObject, 16 ) \
+    SYSCALL_ENTRY( 0x0069, NtGdiGetDCPoint, 24 ) \
+    SYSCALL_ENTRY( 0x006a, NtGdiGetDIBitsInternal, 72 ) \
+    SYSCALL_ENTRY( 0x006b, NtGdiGetDeviceCaps, 16 ) \
+    SYSCALL_ENTRY( 0x006c, NtGdiGetDeviceGammaRamp, 16 ) \
+    SYSCALL_ENTRY( 0x006d, NtGdiGetFontData, 40 ) \
+    SYSCALL_ENTRY( 0x006e, NtGdiGetFontFileData, 40 ) \
+    SYSCALL_ENTRY( 0x006f, NtGdiGetFontFileInfo, 40 ) \
+    SYSCALL_ENTRY( 0x0070, NtGdiGetFontUnicodeRanges, 16 ) \
+    SYSCALL_ENTRY( 0x0071, NtGdiGetGlyphIndicesW, 40 ) \
+    SYSCALL_ENTRY( 0x0072, NtGdiGetGlyphOutline, 64 ) \
+    SYSCALL_ENTRY( 0x0073, NtGdiGetKerningPairs, 24 ) \
+    SYSCALL_ENTRY( 0x0074, NtGdiGetNearestColor, 16 ) \
+    SYSCALL_ENTRY( 0x0075, NtGdiGetNearestPaletteIndex, 16 ) \
+    SYSCALL_ENTRY( 0x0076, NtGdiGetOutlineTextMetricsInternalW, 32 ) \
+    SYSCALL_ENTRY( 0x0077, NtGdiGetPath, 32 ) \
+    SYSCALL_ENTRY( 0x0078, NtGdiGetPixel, 24 ) \
+    SYSCALL_ENTRY( 0x0079, NtGdiGetRandomRgn, 24 ) \
+    SYSCALL_ENTRY( 0x007a, NtGdiGetRasterizerCaps, 16 ) \
+    SYSCALL_ENTRY( 0x007b, NtGdiGetRealizationInfo, 16 ) \
+    SYSCALL_ENTRY( 0x007c, NtGdiGetRegionData, 24 ) \
+    SYSCALL_ENTRY( 0x007d, NtGdiGetRgnBox, 16 ) \
+    SYSCALL_ENTRY( 0x007e, NtGdiGetSpoolMessage, 32 ) \
+    SYSCALL_ENTRY( 0x007f, NtGdiGetSystemPaletteUse, 8 ) \
+    SYSCALL_ENTRY( 0x0080, NtGdiGetTextCharsetInfo, 24 ) \
+    SYSCALL_ENTRY( 0x0081, NtGdiGetTextExtentExW, 64 ) \
+    SYSCALL_ENTRY( 0x0082, NtGdiGetTextFaceW, 32 ) \
+    SYSCALL_ENTRY( 0x0083, NtGdiGetTextMetricsW, 24 ) \
+    SYSCALL_ENTRY( 0x0084, NtGdiGetTransform, 24 ) \
+    SYSCALL_ENTRY( 0x0085, NtGdiGradientFill, 48 ) \
+    SYSCALL_ENTRY( 0x0086, NtGdiHfontCreate, 40 ) \
+    SYSCALL_ENTRY( 0x0087, NtGdiIcmBrushInfo, 64 ) \
+    SYSCALL_ENTRY( 0x0088, NtGdiInitSpool, 0 ) \
+    SYSCALL_ENTRY( 0x0089, NtGdiIntersectClipRect, 40 ) \
+    SYSCALL_ENTRY( 0x008a, NtGdiInvertRgn, 16 ) \
+    SYSCALL_ENTRY( 0x008b, NtGdiLineTo, 24 ) \
+    SYSCALL_ENTRY( 0x008c, NtGdiMaskBlt, 104 ) \
+    SYSCALL_ENTRY( 0x008d, NtGdiModifyWorldTransform, 24 ) \
+    SYSCALL_ENTRY( 0x008e, NtGdiMoveTo, 32 ) \
+    SYSCALL_ENTRY( 0x008f, NtGdiOffsetClipRgn, 24 ) \
+    SYSCALL_ENTRY( 0x0090, NtGdiOffsetRgn, 24 ) \
+    SYSCALL_ENTRY( 0x0091, NtGdiOpenDCW, 64 ) \
+    SYSCALL_ENTRY( 0x0092, NtGdiPatBlt, 48 ) \
+    SYSCALL_ENTRY( 0x0093, NtGdiPathToRegion, 8 ) \
+    SYSCALL_ENTRY( 0x0094, NtGdiPlgBlt, 88 ) \
+    SYSCALL_ENTRY( 0x0095, NtGdiPolyDraw, 32 ) \
+    SYSCALL_ENTRY( 0x0096, NtGdiPolyPolyDraw, 40 ) \
+    SYSCALL_ENTRY( 0x0097, NtGdiPtInRegion, 24 ) \
+    SYSCALL_ENTRY( 0x0098, NtGdiPtVisible, 24 ) \
+    SYSCALL_ENTRY( 0x0099, NtGdiRectInRegion, 16 ) \
+    SYSCALL_ENTRY( 0x009a, NtGdiRectVisible, 16 ) \
+    SYSCALL_ENTRY( 0x009b, NtGdiRectangle, 40 ) \
+    SYSCALL_ENTRY( 0x009c, NtGdiRemoveFontMemResourceEx, 8 ) \
+    SYSCALL_ENTRY( 0x009d, NtGdiRemoveFontResourceW, 48 ) \
+    SYSCALL_ENTRY( 0x009e, NtGdiResetDC, 40 ) \
+    SYSCALL_ENTRY( 0x009f, NtGdiResizePalette, 16 ) \
+    SYSCALL_ENTRY( 0x00a0, NtGdiRestoreDC, 16 ) \
+    SYSCALL_ENTRY( 0x00a1, NtGdiRoundRect, 56 ) \
+    SYSCALL_ENTRY( 0x00a2, NtGdiSaveDC, 8 ) \
+    SYSCALL_ENTRY( 0x00a3, NtGdiScaleViewportExtEx, 48 ) \
+    SYSCALL_ENTRY( 0x00a4, NtGdiScaleWindowExtEx, 48 ) \
+    SYSCALL_ENTRY( 0x00a5, NtGdiSelectBitmap, 16 ) \
+    SYSCALL_ENTRY( 0x00a6, NtGdiSelectBrush, 16 ) \
+    SYSCALL_ENTRY( 0x00a7, NtGdiSelectClipPath, 16 ) \
+    SYSCALL_ENTRY( 0x00a8, NtGdiSelectFont, 16 ) \
+    SYSCALL_ENTRY( 0x00a9, NtGdiSelectPen, 16 ) \
+    SYSCALL_ENTRY( 0x00aa, NtGdiSetBitmapBits, 24 ) \
+    SYSCALL_ENTRY( 0x00ab, NtGdiSetBitmapDimension, 32 ) \
+    SYSCALL_ENTRY( 0x00ac, NtGdiSetBoundsRect, 24 ) \
+    SYSCALL_ENTRY( 0x00ad, NtGdiSetBrushOrg, 32 ) \
+    SYSCALL_ENTRY( 0x00ae, NtGdiSetColorAdjustment, 16 ) \
+    SYSCALL_ENTRY( 0x00af, NtGdiSetDIBitsToDeviceInternal, 128 ) \
+    SYSCALL_ENTRY( 0x00b0, NtGdiSetDeviceGammaRamp, 16 ) \
+    SYSCALL_ENTRY( 0x00b1, NtGdiSetLayout, 24 ) \
+    SYSCALL_ENTRY( 0x00b2, NtGdiSetMagicColors, 24 ) \
+    SYSCALL_ENTRY( 0x00b3, NtGdiSetMetaRgn, 8 ) \
+    SYSCALL_ENTRY( 0x00b4, NtGdiSetPixel, 32 ) \
+    SYSCALL_ENTRY( 0x00b5, NtGdiSetPixelFormat, 16 ) \
+    SYSCALL_ENTRY( 0x00b6, NtGdiSetRectRgn, 40 ) \
+    SYSCALL_ENTRY( 0x00b7, NtGdiSetSystemPaletteUse, 16 ) \
+    SYSCALL_ENTRY( 0x00b8, NtGdiSetTextJustification, 24 ) \
+    SYSCALL_ENTRY( 0x00b9, NtGdiSetVirtualResolution, 40 ) \
+    SYSCALL_ENTRY( 0x00ba, NtGdiStartDoc, 32 ) \
+    SYSCALL_ENTRY( 0x00bb, NtGdiStartPage, 8 ) \
+    SYSCALL_ENTRY( 0x00bc, NtGdiStretchBlt, 96 ) \
+    SYSCALL_ENTRY( 0x00bd, NtGdiStretchDIBitsInternal, 128 ) \
+    SYSCALL_ENTRY( 0x00be, NtGdiStrokeAndFillPath, 8 ) \
+    SYSCALL_ENTRY( 0x00bf, NtGdiStrokePath, 8 ) \
+    SYSCALL_ENTRY( 0x00c0, NtGdiSwapBuffers, 8 ) \
+    SYSCALL_ENTRY( 0x00c1, NtGdiTransformPoints, 40 ) \
+    SYSCALL_ENTRY( 0x00c2, NtGdiTransparentBlt, 88 ) \
+    SYSCALL_ENTRY( 0x00c3, NtGdiUnrealizeObject, 8 ) \
+    SYSCALL_ENTRY( 0x00c4, NtGdiUpdateColors, 8 ) \
+    SYSCALL_ENTRY( 0x00c5, NtGdiWidenPath, 8 ) \
+    SYSCALL_ENTRY( 0x00c6, NtUserActivateKeyboardLayout, 16 ) \
+    SYSCALL_ENTRY( 0x00c7, NtUserAddClipboardFormatListener, 8 ) \
+    SYSCALL_ENTRY( 0x00c8, NtUserAssociateInputContext, 24 ) \
+    SYSCALL_ENTRY( 0x00c9, NtUserAttachThreadInput, 24 ) \
+    SYSCALL_ENTRY( 0x00ca, NtUserBeginPaint, 16 ) \
+    SYSCALL_ENTRY( 0x00cb, NtUserBuildHimcList, 32 ) \
+    SYSCALL_ENTRY( 0x00cc, NtUserBuildHwndList, 64 ) \
+    SYSCALL_ENTRY( 0x00cd, NtUserBuildNameList, 32 ) \
+    SYSCALL_ENTRY( 0x00ce, NtUserBuildPropList, 32 ) \
+    SYSCALL_ENTRY( 0x00cf, NtUserCallHwnd, 16 ) \
+    SYSCALL_ENTRY( 0x00d0, NtUserCallHwndParam, 24 ) \
+    SYSCALL_ENTRY( 0x00d1, NtUserCallMsgFilter, 16 ) \
+    SYSCALL_ENTRY( 0x00d2, NtUserCallNextHookEx, 32 ) \
+    SYSCALL_ENTRY( 0x00d3, NtUserCallNoParam, 8 ) \
+    SYSCALL_ENTRY( 0x00d4, NtUserCallOneParam, 16 ) \
+    SYSCALL_ENTRY( 0x00d5, NtUserCallTwoParam, 24 ) \
+    SYSCALL_ENTRY( 0x00d6, NtUserChangeClipboardChain, 16 ) \
+    SYSCALL_ENTRY( 0x00d7, NtUserChangeDisplaySettings, 40 ) \
+    SYSCALL_ENTRY( 0x00d8, NtUserCheckMenuItem, 24 ) \
+    SYSCALL_ENTRY( 0x00d9, NtUserChildWindowFromPointEx, 32 ) \
+    SYSCALL_ENTRY( 0x00da, NtUserClipCursor, 8 ) \
+    SYSCALL_ENTRY( 0x00db, NtUserCloseClipboard, 0 ) \
+    SYSCALL_ENTRY( 0x00dc, NtUserCloseDesktop, 8 ) \
+    SYSCALL_ENTRY( 0x00dd, NtUserCloseWindowStation, 8 ) \
+    SYSCALL_ENTRY( 0x00de, NtUserCopyAcceleratorTable, 24 ) \
+    SYSCALL_ENTRY( 0x00df, NtUserCountClipboardFormats, 0 ) \
+    SYSCALL_ENTRY( 0x00e0, NtUserCreateAcceleratorTable, 16 ) \
+    SYSCALL_ENTRY( 0x00e1, NtUserCreateCaret, 32 ) \
+    SYSCALL_ENTRY( 0x00e2, NtUserCreateDesktopEx, 48 ) \
+    SYSCALL_ENTRY( 0x00e3, NtUserCreateInputContext, 8 ) \
+    SYSCALL_ENTRY( 0x00e4, NtUserCreateMenu, 0 ) \
+    SYSCALL_ENTRY( 0x00e5, NtUserCreatePopupMenu, 0 ) \
+    SYSCALL_ENTRY( 0x00e6, NtUserCreateWindowEx, 136 ) \
+    SYSCALL_ENTRY( 0x00e7, NtUserCreateWindowStation, 56 ) \
+    SYSCALL_ENTRY( 0x00e8, NtUserDeferWindowPosAndBand, 80 ) \
+    SYSCALL_ENTRY( 0x00e9, NtUserDeleteMenu, 24 ) \
+    SYSCALL_ENTRY( 0x00ea, NtUserDestroyAcceleratorTable, 8 ) \
+    SYSCALL_ENTRY( 0x00eb, NtUserDestroyCaret, 0 ) \
+    SYSCALL_ENTRY( 0x00ec, NtUserDestroyCursor, 16 ) \
+    SYSCALL_ENTRY( 0x00ed, NtUserDestroyInputContext, 8 ) \
+    SYSCALL_ENTRY( 0x00ee, NtUserDestroyMenu, 8 ) \
+    SYSCALL_ENTRY( 0x00ef, NtUserDestroyWindow, 8 ) \
+    SYSCALL_ENTRY( 0x00f0, NtUserDisableThreadIme, 8 ) \
+    SYSCALL_ENTRY( 0x00f1, NtUserDispatchMessage, 8 ) \
+    SYSCALL_ENTRY( 0x00f2, NtUserDisplayConfigGetDeviceInfo, 8 ) \
+    SYSCALL_ENTRY( 0x00f3, NtUserDragDetect, 24 ) \
+    SYSCALL_ENTRY( 0x00f4, NtUserDragObject, 40 ) \
+    SYSCALL_ENTRY( 0x00f5, NtUserDrawCaptionTemp, 56 ) \
+    SYSCALL_ENTRY( 0x00f6, NtUserDrawIconEx, 72 ) \
+    SYSCALL_ENTRY( 0x00f7, NtUserDrawMenuBarTemp, 40 ) \
+    SYSCALL_ENTRY( 0x00f8, NtUserEmptyClipboard, 0 ) \
+    SYSCALL_ENTRY( 0x00f9, NtUserEnableMenuItem, 24 ) \
+    SYSCALL_ENTRY( 0x00fa, NtUserEnableMouseInPointer, 8 ) \
+    SYSCALL_ENTRY( 0x00fb, NtUserEnableMouseInPointerForThread, 0 ) \
+    SYSCALL_ENTRY( 0x00fc, NtUserEnableScrollBar, 24 ) \
+    SYSCALL_ENTRY( 0x00fd, NtUserEndDeferWindowPosEx, 16 ) \
+    SYSCALL_ENTRY( 0x00fe, NtUserEndMenu, 0 ) \
+    SYSCALL_ENTRY( 0x00ff, NtUserEndPaint, 16 ) \
+    SYSCALL_ENTRY( 0x0100, NtUserEnumClipboardFormats, 8 ) \
+    SYSCALL_ENTRY( 0x0101, NtUserEnumDisplayDevices, 32 ) \
+    SYSCALL_ENTRY( 0x0102, NtUserEnumDisplayMonitors, 32 ) \
+    SYSCALL_ENTRY( 0x0103, NtUserEnumDisplaySettings, 32 ) \
+    SYSCALL_ENTRY( 0x0104, NtUserExcludeUpdateRgn, 16 ) \
+    SYSCALL_ENTRY( 0x0105, NtUserFindExistingCursorIcon, 24 ) \
+    SYSCALL_ENTRY( 0x0106, NtUserFindWindowEx, 40 ) \
+    SYSCALL_ENTRY( 0x0107, NtUserFlashWindowEx, 8 ) \
+    SYSCALL_ENTRY( 0x0108, NtUserGetAncestor, 16 ) \
+    SYSCALL_ENTRY( 0x0109, NtUserGetAsyncKeyState, 8 ) \
+    SYSCALL_ENTRY( 0x010a, NtUserGetAtomName, 16 ) \
+    SYSCALL_ENTRY( 0x010b, NtUserGetCaretBlinkTime, 0 ) \
+    SYSCALL_ENTRY( 0x010c, NtUserGetCaretPos, 8 ) \
+    SYSCALL_ENTRY( 0x010d, NtUserGetClassInfoEx, 40 ) \
+    SYSCALL_ENTRY( 0x010e, NtUserGetClassName, 24 ) \
+    SYSCALL_ENTRY( 0x010f, NtUserGetClipboardData, 16 ) \
+    SYSCALL_ENTRY( 0x0110, NtUserGetClipboardFormatName, 24 ) \
+    SYSCALL_ENTRY( 0x0111, NtUserGetClipboardOwner, 0 ) \
+    SYSCALL_ENTRY( 0x0112, NtUserGetClipboardSequenceNumber, 0 ) \
+    SYSCALL_ENTRY( 0x0113, NtUserGetClipboardViewer, 0 ) \
+    SYSCALL_ENTRY( 0x0114, NtUserGetCurrentInputMessageSource, 8 ) \
+    SYSCALL_ENTRY( 0x0115, NtUserGetCursor, 0 ) \
+    SYSCALL_ENTRY( 0x0116, NtUserGetCursorFrameInfo, 32 ) \
+    SYSCALL_ENTRY( 0x0117, NtUserGetCursorInfo, 8 ) \
+    SYSCALL_ENTRY( 0x0118, NtUserGetDC, 8 ) \
+    SYSCALL_ENTRY( 0x0119, NtUserGetDCEx, 24 ) \
+    SYSCALL_ENTRY( 0x011a, NtUserGetDisplayConfigBufferSizes, 24 ) \
+    SYSCALL_ENTRY( 0x011b, NtUserGetDoubleClickTime, 0 ) \
+    SYSCALL_ENTRY( 0x011c, NtUserGetDpiForMonitor, 32 ) \
+    SYSCALL_ENTRY( 0x011d, NtUserGetForegroundWindow, 0 ) \
+    SYSCALL_ENTRY( 0x011e, NtUserGetGUIThreadInfo, 16 ) \
+    SYSCALL_ENTRY( 0x011f, NtUserGetIconInfo, 48 ) \
+    SYSCALL_ENTRY( 0x0120, NtUserGetIconSize, 32 ) \
+    SYSCALL_ENTRY( 0x0121, NtUserGetInternalWindowPos, 24 ) \
+    SYSCALL_ENTRY( 0x0122, NtUserGetKeyNameText, 24 ) \
+    SYSCALL_ENTRY( 0x0123, NtUserGetKeyState, 8 ) \
+    SYSCALL_ENTRY( 0x0124, NtUserGetKeyboardLayout, 8 ) \
+    SYSCALL_ENTRY( 0x0125, NtUserGetKeyboardLayoutList, 16 ) \
+    SYSCALL_ENTRY( 0x0126, NtUserGetKeyboardLayoutName, 8 ) \
+    SYSCALL_ENTRY( 0x0127, NtUserGetKeyboardState, 8 ) \
+    SYSCALL_ENTRY( 0x0128, NtUserGetLayeredWindowAttributes, 32 ) \
+    SYSCALL_ENTRY( 0x0129, NtUserGetMenuBarInfo, 32 ) \
+    SYSCALL_ENTRY( 0x012a, NtUserGetMenuItemRect, 32 ) \
+    SYSCALL_ENTRY( 0x012b, NtUserGetMessage, 32 ) \
+    SYSCALL_ENTRY( 0x012c, NtUserGetMouseMovePointsEx, 40 ) \
+    SYSCALL_ENTRY( 0x012d, NtUserGetObjectInformation, 40 ) \
+    SYSCALL_ENTRY( 0x012e, NtUserGetOpenClipboardWindow, 0 ) \
+    SYSCALL_ENTRY( 0x012f, NtUserGetPointerInfoList, 64 ) \
+    SYSCALL_ENTRY( 0x0130, NtUserGetPriorityClipboardFormat, 16 ) \
+    SYSCALL_ENTRY( 0x0131, NtUserGetProcessDpiAwarenessContext, 8 ) \
+    SYSCALL_ENTRY( 0x0132, NtUserGetProcessWindowStation, 0 ) \
+    SYSCALL_ENTRY( 0x0133, NtUserGetProp, 16 ) \
+    SYSCALL_ENTRY( 0x0134, NtUserGetQueueStatus, 8 ) \
+    SYSCALL_ENTRY( 0x0135, NtUserGetRawInputBuffer, 24 ) \
+    SYSCALL_ENTRY( 0x0136, NtUserGetRawInputData, 40 ) \
+    SYSCALL_ENTRY( 0x0137, NtUserGetRawInputDeviceInfo, 32 ) \
+    SYSCALL_ENTRY( 0x0138, NtUserGetRawInputDeviceList, 24 ) \
+    SYSCALL_ENTRY( 0x0139, NtUserGetRegisteredRawInputDevices, 24 ) \
+    SYSCALL_ENTRY( 0x013a, NtUserGetScrollBarInfo, 24 ) \
+    SYSCALL_ENTRY( 0x013b, NtUserGetSystemDpiForProcess, 8 ) \
+    SYSCALL_ENTRY( 0x013c, NtUserGetSystemMenu, 16 ) \
+    SYSCALL_ENTRY( 0x013d, NtUserGetThreadDesktop, 8 ) \
+    SYSCALL_ENTRY( 0x013e, NtUserGetThreadState, 8 ) \
+    SYSCALL_ENTRY( 0x013f, NtUserGetTitleBarInfo, 16 ) \
+    SYSCALL_ENTRY( 0x0140, NtUserGetTouchInputInfo, 32 ) \
+    SYSCALL_ENTRY( 0x0141, NtUserGetUpdateRect, 24 ) \
+    SYSCALL_ENTRY( 0x0142, NtUserGetUpdateRgn, 24 ) \
+    SYSCALL_ENTRY( 0x0143, NtUserGetUpdatedClipboardFormats, 24 ) \
+    SYSCALL_ENTRY( 0x0144, NtUserGetWindowDC, 8 ) \
+    SYSCALL_ENTRY( 0x0145, NtUserGetWindowPlacement, 16 ) \
+    SYSCALL_ENTRY( 0x0146, NtUserGetWindowRgnEx, 24 ) \
+    SYSCALL_ENTRY( 0x0147, NtUserHideCaret, 8 ) \
+    SYSCALL_ENTRY( 0x0148, NtUserHiliteMenuItem, 32 ) \
+    SYSCALL_ENTRY( 0x0149, NtUserInitializeClientPfnArrays, 32 ) \
+    SYSCALL_ENTRY( 0x014a, NtUserInternalGetWindowIcon, 16 ) \
+    SYSCALL_ENTRY( 0x014b, NtUserInternalGetWindowText, 24 ) \
+    SYSCALL_ENTRY( 0x014c, NtUserInvalidateRect, 24 ) \
+    SYSCALL_ENTRY( 0x014d, NtUserInvalidateRgn, 24 ) \
+    SYSCALL_ENTRY( 0x014e, NtUserIsClipboardFormatAvailable, 8 ) \
+    SYSCALL_ENTRY( 0x014f, NtUserIsMouseInPointerEnabled, 0 ) \
+    SYSCALL_ENTRY( 0x0150, NtUserIsTouchWindow, 16 ) \
+    SYSCALL_ENTRY( 0x0151, NtUserKillTimer, 16 ) \
+    SYSCALL_ENTRY( 0x0152, NtUserLockWindowUpdate, 8 ) \
+    SYSCALL_ENTRY( 0x0153, NtUserLogicalToPerMonitorDPIPhysicalPoint, 16 ) \
+    SYSCALL_ENTRY( 0x0154, NtUserMapVirtualKeyEx, 24 ) \
+    SYSCALL_ENTRY( 0x0155, NtUserMenuItemFromPoint, 32 ) \
+    SYSCALL_ENTRY( 0x0156, NtUserMessageBeep, 8 ) \
+    SYSCALL_ENTRY( 0x0157, NtUserMessageCall, 56 ) \
+    SYSCALL_ENTRY( 0x0158, NtUserMoveWindow, 48 ) \
+    SYSCALL_ENTRY( 0x0159, NtUserMsgWaitForMultipleObjectsEx, 40 ) \
+    SYSCALL_ENTRY( 0x015a, NtUserNotifyIMEStatus, 16 ) \
+    SYSCALL_ENTRY( 0x015b, NtUserNotifyWinEvent, 32 ) \
+    SYSCALL_ENTRY( 0x015c, NtUserOpenClipboard, 16 ) \
+    SYSCALL_ENTRY( 0x015d, NtUserOpenDesktop, 24 ) \
+    SYSCALL_ENTRY( 0x015e, NtUserOpenInputDesktop, 24 ) \
+    SYSCALL_ENTRY( 0x015f, NtUserOpenWindowStation, 16 ) \
+    SYSCALL_ENTRY( 0x0160, NtUserPeekMessage, 40 ) \
+    SYSCALL_ENTRY( 0x0161, NtUserPerMonitorDPIPhysicalToLogicalPoint, 16 ) \
+    SYSCALL_ENTRY( 0x0162, NtUserPostMessage, 32 ) \
+    SYSCALL_ENTRY( 0x0163, NtUserPostQuitMessage, 8 ) \
+    SYSCALL_ENTRY( 0x0164, NtUserPostThreadMessage, 32 ) \
+    SYSCALL_ENTRY( 0x0165, NtUserPrintWindow, 24 ) \
+    SYSCALL_ENTRY( 0x0166, NtUserQueryDisplayConfig, 48 ) \
+    SYSCALL_ENTRY( 0x0167, NtUserQueryInputContext, 16 ) \
+    SYSCALL_ENTRY( 0x0168, NtUserQueryWindow, 16 ) \
+    SYSCALL_ENTRY( 0x0169, NtUserRealChildWindowFromPoint, 24 ) \
+    SYSCALL_ENTRY( 0x016a, NtUserRedrawWindow, 32 ) \
+    SYSCALL_ENTRY( 0x016b, NtUserRegisterClassExWOW, 56 ) \
+    SYSCALL_ENTRY( 0x016c, NtUserRegisterHotKey, 32 ) \
+    SYSCALL_ENTRY( 0x016d, NtUserRegisterRawInputDevices, 24 ) \
+    SYSCALL_ENTRY( 0x016e, NtUserRegisterTouchPadCapable, 8 ) \
+    SYSCALL_ENTRY( 0x016f, NtUserReleaseCapture, 0 ) \
+    SYSCALL_ENTRY( 0x0170, NtUserReleaseDC, 16 ) \
+    SYSCALL_ENTRY( 0x0171, NtUserRemoveClipboardFormatListener, 8 ) \
+    SYSCALL_ENTRY( 0x0172, NtUserRemoveMenu, 24 ) \
+    SYSCALL_ENTRY( 0x0173, NtUserRemoveProp, 16 ) \
+    SYSCALL_ENTRY( 0x0174, NtUserScrollDC, 56 ) \
+    SYSCALL_ENTRY( 0x0175, NtUserScrollWindowEx, 64 ) \
+    SYSCALL_ENTRY( 0x0176, NtUserSelectPalette, 24 ) \
+    SYSCALL_ENTRY( 0x0177, NtUserSendInput, 24 ) \
+    SYSCALL_ENTRY( 0x0178, NtUserSetActiveWindow, 8 ) \
+    SYSCALL_ENTRY( 0x0179, NtUserSetAdditionalForegroundBoostProcesses, 24 ) \
+    SYSCALL_ENTRY( 0x017a, NtUserSetCapture, 8 ) \
+    SYSCALL_ENTRY( 0x017b, NtUserSetClassLong, 32 ) \
+    SYSCALL_ENTRY( 0x017c, NtUserSetClassLongPtr, 32 ) \
+    SYSCALL_ENTRY( 0x017d, NtUserSetClassWord, 24 ) \
+    SYSCALL_ENTRY( 0x017e, NtUserSetClipboardData, 24 ) \
+    SYSCALL_ENTRY( 0x017f, NtUserSetClipboardViewer, 8 ) \
+    SYSCALL_ENTRY( 0x0180, NtUserSetCursor, 8 ) \
+    SYSCALL_ENTRY( 0x0181, NtUserSetCursorIconData, 32 ) \
+    SYSCALL_ENTRY( 0x0182, NtUserSetCursorPos, 16 ) \
+    SYSCALL_ENTRY( 0x0183, NtUserSetFocus, 8 ) \
+    SYSCALL_ENTRY( 0x0184, NtUserSetInternalWindowPos, 32 ) \
+    SYSCALL_ENTRY( 0x0185, NtUserSetKeyboardState, 8 ) \
+    SYSCALL_ENTRY( 0x0186, NtUserSetLayeredWindowAttributes, 32 ) \
+    SYSCALL_ENTRY( 0x0187, NtUserSetMenu, 16 ) \
+    SYSCALL_ENTRY( 0x0188, NtUserSetMenuContextHelpId, 16 ) \
+    SYSCALL_ENTRY( 0x0189, NtUserSetMenuDefaultItem, 24 ) \
+    SYSCALL_ENTRY( 0x018a, NtUserSetObjectInformation, 32 ) \
+    SYSCALL_ENTRY( 0x018b, NtUserSetParent, 16 ) \
+    SYSCALL_ENTRY( 0x018c, NtUserSetProcessDpiAwarenessContext, 16 ) \
+    SYSCALL_ENTRY( 0x018d, NtUserSetProcessWindowStation, 8 ) \
+    SYSCALL_ENTRY( 0x018e, NtUserSetProp, 24 ) \
+    SYSCALL_ENTRY( 0x018f, NtUserSetScrollInfo, 32 ) \
+    SYSCALL_ENTRY( 0x0190, NtUserSetShellWindowEx, 16 ) \
+    SYSCALL_ENTRY( 0x0191, NtUserSetSysColors, 24 ) \
+    SYSCALL_ENTRY( 0x0192, NtUserSetSystemMenu, 16 ) \
+    SYSCALL_ENTRY( 0x0193, NtUserSetSystemTimer, 24 ) \
+    SYSCALL_ENTRY( 0x0194, NtUserSetThreadDesktop, 8 ) \
+    SYSCALL_ENTRY( 0x0195, NtUserSetTimer, 40 ) \
+    SYSCALL_ENTRY( 0x0196, NtUserSetWinEventHook, 64 ) \
+    SYSCALL_ENTRY( 0x0197, NtUserSetWindowLong, 32 ) \
+    SYSCALL_ENTRY( 0x0198, NtUserSetWindowLongPtr, 32 ) \
+    SYSCALL_ENTRY( 0x0199, NtUserSetWindowPlacement, 16 ) \
+    SYSCALL_ENTRY( 0x019a, NtUserSetWindowPos, 56 ) \
+    SYSCALL_ENTRY( 0x019b, NtUserSetWindowRgn, 24 ) \
+    SYSCALL_ENTRY( 0x019c, NtUserSetWindowWord, 24 ) \
+    SYSCALL_ENTRY( 0x019d, NtUserSetWindowsHookEx, 48 ) \
+    SYSCALL_ENTRY( 0x019e, NtUserShowCaret, 8 ) \
+    SYSCALL_ENTRY( 0x019f, NtUserShowCursor, 8 ) \
+    SYSCALL_ENTRY( 0x01a0, NtUserShowScrollBar, 24 ) \
+    SYSCALL_ENTRY( 0x01a1, NtUserShowWindow, 16 ) \
+    SYSCALL_ENTRY( 0x01a2, NtUserShowWindowAsync, 16 ) \
+    SYSCALL_ENTRY( 0x01a3, NtUserSwitchDesktop, 8 ) \
+    SYSCALL_ENTRY( 0x01a4, NtUserSystemParametersInfo, 32 ) \
+    SYSCALL_ENTRY( 0x01a5, NtUserSystemParametersInfoForDpi, 40 ) \
+    SYSCALL_ENTRY( 0x01a6, NtUserThunkedMenuInfo, 16 ) \
+    SYSCALL_ENTRY( 0x01a7, NtUserThunkedMenuItemInfo, 48 ) \
+    SYSCALL_ENTRY( 0x01a8, NtUserToUnicodeEx, 56 ) \
+    SYSCALL_ENTRY( 0x01a9, NtUserTrackMouseEvent, 8 ) \
+    SYSCALL_ENTRY( 0x01aa, NtUserTrackPopupMenuEx, 48 ) \
+    SYSCALL_ENTRY( 0x01ab, NtUserTranslateAccelerator, 24 ) \
+    SYSCALL_ENTRY( 0x01ac, NtUserTranslateMessage, 16 ) \
+    SYSCALL_ENTRY( 0x01ad, NtUserUnhookWinEvent, 8 ) \
+    SYSCALL_ENTRY( 0x01ae, NtUserUnhookWindowsHookEx, 8 ) \
+    SYSCALL_ENTRY( 0x01af, NtUserUnregisterClass, 24 ) \
+    SYSCALL_ENTRY( 0x01b0, NtUserUnregisterHotKey, 16 ) \
+    SYSCALL_ENTRY( 0x01b1, NtUserUpdateInputContext, 24 ) \
+    SYSCALL_ENTRY( 0x01b2, NtUserUpdateLayeredWindow, 80 ) \
+    SYSCALL_ENTRY( 0x01b3, NtUserValidateRect, 16 ) \
+    SYSCALL_ENTRY( 0x01b4, NtUserVkKeyScanEx, 16 ) \
+    SYSCALL_ENTRY( 0x01b5, NtUserWaitForInputIdle, 24 ) \
+    SYSCALL_ENTRY( 0x01b6, NtUserWaitMessage, 0 ) \
+    SYSCALL_ENTRY( 0x01b7, NtUserWindowFromDC, 8 ) \
+    SYSCALL_ENTRY( 0x01b8, NtUserWindowFromPoint, 16 ) \
+    SYSCALL_ENTRY( 0x01b9, __wine_get_file_outline_text_metric, 32 ) \
+    SYSCALL_ENTRY( 0x01ba, __wine_get_icm_profile, 32 )
diff --git a/dlls/win32u/win32u.spec b/dlls/win32u/win32u.spec
index 417d5e017c1..f4f1513df95 100644
--- a/dlls/win32u/win32u.spec
+++ b/dlls/win32u/win32u.spec
@@ -1330,7 +1330,7 @@
 @ stub NtUserSetActivationFilter
 @ stub NtUserSetActiveProcessForMonitor
 @ stdcall -syscall NtUserSetActiveWindow(long)
-@ stub NtUserSetAdditionalForegroundBoostProcesses
+@ stdcall -syscall NtUserSetAdditionalForegroundBoostProcesses(ptr long ptr)
 @ stub NtUserSetAppImeLevel
 @ stub NtUserSetAutoRotation
 @ stub NtUserSetBridgeWindowChild
diff --git a/dlls/wow64win/user.c b/dlls/wow64win/user.c
index effe8ebffae..0d2cc9cb7a7 100644
--- a/dlls/wow64win/user.c
+++ b/dlls/wow64win/user.c
@@ -4129,6 +4129,25 @@ NTSTATUS WINAPI wow64_NtUserSetActiveWindow( UINT *args )
     return HandleToUlong( NtUserSetActiveWindow( hwnd ));
 }
 
+NTSTATUS WINAPI wow64_NtUserSetAdditionalForegroundBoostProcesses( UINT *args )
+{
+    HWND hwnd = get_handle( &args );
+    DWORD count = get_ulong( &args );
+    ULONG *handles32 = get_ptr( &args );
+
+    HANDLE handles[32];
+    unsigned int i;
+
+    if (count > ARRAYSIZE(handles))
+    {
+        set_last_error32( ERROR_INVALID_PARAMETER );
+        return FALSE;
+    }
+    for (i = 0; i < count; i++) handles[i] = LongToHandle( handles32[i] );
+
+    return NtUserSetAdditionalForegroundBoostProcesses( hwnd, count, handles );
+}
+
 NTSTATUS WINAPI wow64_NtUserSetCapture( UINT *args )
 {
     HWND hwnd = get_handle( &args );
diff --git a/include/ntuser.h b/include/ntuser.h
index 6f5c0eba0ee..92fa0dec76b 100644
--- a/include/ntuser.h
+++ b/include/ntuser.h
@@ -931,6 +931,7 @@ W32KAPI INT     WINAPI NtUserScrollWindowEx( HWND hwnd, INT dx, INT dy, const RE
 W32KAPI HPALETTE WINAPI NtUserSelectPalette( HDC hdc, HPALETTE palette, WORD force_background );
 W32KAPI UINT     WINAPI NtUserSendInput( UINT count, INPUT *inputs, int size );
 W32KAPI HWND     WINAPI NtUserSetActiveWindow( HWND hwnd );
+W32KAPI BOOL     WINAPI NtUserSetAdditionalForegroundBoostProcesses( HWND hwnd, DWORD count, HANDLE *handles );
 W32KAPI HWND     WINAPI NtUserSetCapture( HWND hwnd );
 W32KAPI DWORD    WINAPI NtUserSetClassLong( HWND hwnd, INT offset, LONG newval, BOOL ansi );
 W32KAPI ULONG_PTR WINAPI NtUserSetClassLongPtr( HWND hwnd, INT offset, LONG_PTR newval, BOOL ansi );
-- 
2.51.0

